"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[522],{3905:function(e,t,n){n.d(t,{Zo:function(){return d},kt:function(){return h}});var a=n(7294);function r(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function i(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){r(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,r=function(e,t){if(null==e)return{};var n,a,r={},o=Object.keys(e);for(a=0;a<o.length;a++)n=o[a],t.indexOf(n)>=0||(r[n]=e[n]);return r}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(a=0;a<o.length;a++)n=o[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(r[n]=e[n])}return r}var l=a.createContext({}),c=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):i(i({},t),e)),n},d=function(e){var t=c(e.components);return a.createElement(l.Provider,{value:t},e.children)},p={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},u=a.forwardRef((function(e,t){var n=e.components,r=e.mdxType,o=e.originalType,l=e.parentName,d=s(e,["components","mdxType","originalType","parentName"]),u=c(n),h=r,m=u["".concat(l,".").concat(h)]||u[h]||p[h]||o;return n?a.createElement(m,i(i({ref:t},d),{},{components:n})):a.createElement(m,i({ref:t},d))}));function h(e,t){var n=arguments,r=t&&t.mdxType;if("string"==typeof e||r){var o=n.length,i=new Array(o);i[0]=u;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s.mdxType="string"==typeof e?e:r,i[1]=s;for(var c=2;c<o;c++)i[c]=n[c];return a.createElement.apply(null,i)}return a.createElement.apply(null,n)}u.displayName="MDXCreateElement"},9528:function(e,t,n){n.r(t),n.d(t,{assets:function(){return d},contentTitle:function(){return l},default:function(){return h},frontMatter:function(){return s},metadata:function(){return c},toc:function(){return p}});var a=n(7462),r=n(3366),o=(n(7294),n(3905)),i=["components"],s={},l="Translation",c={unversionedId:"biological-foundations/translation",id:"biological-foundations/translation",title:"Translation",description:"Decoding the Transcript",source:"@site/docs/01-biological-foundations/04-translation.md",sourceDirName:"01-biological-foundations",slug:"/biological-foundations/translation",permalink:"/docs/biological-foundations/translation",editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/01-biological-foundations/04-translation.md",tags:[],version:"current",sidebarPosition:4,frontMatter:{},sidebar:"tutorialSidebar",previous:{title:"Transcription",permalink:"/docs/biological-foundations/transcription"},next:{title:"Genotype, Phenotype and Haplotype",permalink:"/docs/biological-foundations/genotype-phenotype-haplotype"}},d={},p=[{value:"Decoding the Transcript",id:"decoding-the-transcript",level:2},{value:"Reading Frames",id:"reading-frames",level:2}],u={toc:p};function h(e){var t=e.components,s=(0,r.Z)(e,i);return(0,o.kt)("wrapper",(0,a.Z)({},u,s,{components:t,mdxType:"MDXLayout"}),(0,o.kt)("h1",{id:"translation"},"Translation"),(0,o.kt)("h2",{id:"decoding-the-transcript"},"Decoding the Transcript"),(0,o.kt)("p",null,"After DNA is transcribed to RNA and the RNA is spliced, the next step in protein\nproduction is translation. The genetic code that cells use to translate RNA to protein\nis universally preserved across organisms. So you, your dog, and the lettuce in your\nsalad all read and translate the genetic code the same. This helps scientists determine\nwhich changes in a gene have functional consequences by comparing transcripts from\ndifferent species. Further, the universality of this code makes modeling genetic\ndiseases in animals\u2014such as grafting cancer onto a mouse, known as a\n",(0,o.kt)("strong",{parentName:"p"},"xenograft"),"\u2014informative."),(0,o.kt)("p",null,(0,o.kt)("img",{alt:"Figure showing translation from mRNA to a protein",src:n(9690).Z,width:"2410",height:"1077"})),(0,o.kt)("p",null,"The mRNA transcript is encoded in three-letter nucleotide blocks known as\xa0",(0,o.kt)("strong",{parentName:"p"},"codons"),".\nCodons specify that a particular protein building block, or ",(0,o.kt)("strong",{parentName:"p"},"amino acid"),', should be\nadded to a growing protein chain. For example, the mRNA sequence above has "CCU" in the\nsecond codon position. This codon instructs the protein assembling machinery of the cell\nto a Proline to the amino acid sequence at position two. '),(0,o.kt)("p",null,(0,o.kt)("img",{alt:"Codon lookup table from mRNA nucleotide triplet",src:n(9355).Z,width:"1178",height:"1069"})),(0,o.kt)("p",null,'As shown in the amino acid lookup table above, other codons specify that translation\nshould start or end at a particular spot. Because multiple triplets result in the same\namino acid, some substitutions may not change which amino acid is added to the chain\n(e.g., if your triplet starts with "CG", the amino acid is always an arginine and the\nthird nucleotide doesn\'t matter). This class of variation, where the mRNA sequence change\nbut the amino acid sequence stays the same, are called "silent mutations". When codon\nvariants change the amino acid, the effect ranges from insignificant to deleterious.\nComputational biologists use programs that incorporate knowledge of the codons and\nchemistry of proteins to predict the effect of a given variant.'),(0,o.kt)("p",null,"As a short aside, DNA and RNA use different coding alphabets. That is, in a mRNA\ntranscript, the DNA base thymine (T) is substituted by uracil (U)."),(0,o.kt)("table",null,(0,o.kt)("thead",{parentName:"table"},(0,o.kt)("tr",{parentName:"thead"},(0,o.kt)("th",{parentName:"tr",align:null},(0,o.kt)("strong",{parentName:"th"},"Molecule")),(0,o.kt)("th",{parentName:"tr",align:null},(0,o.kt)("strong",{parentName:"th"},"Alphabet")))),(0,o.kt)("tbody",{parentName:"table"},(0,o.kt)("tr",{parentName:"tbody"},(0,o.kt)("td",{parentName:"tr",align:null},"DNA"),(0,o.kt)("td",{parentName:"tr",align:null},"ACGT")),(0,o.kt)("tr",{parentName:"tbody"},(0,o.kt)("td",{parentName:"tr",align:null},"RNA"),(0,o.kt)("td",{parentName:"tr",align:null},"ACGU (Uracil replaces Thymine, i.e., U replaces T)")),(0,o.kt)("tr",{parentName:"tbody"},(0,o.kt)("td",{parentName:"tr",align:null},"Proteins"),(0,o.kt)("td",{parentName:"tr",align:null},"Uses a series of triplets in RNA (codons) to encode an amino acid sequence")))),(0,o.kt)("h2",{id:"reading-frames"},"Reading Frames"),(0,o.kt)("p",null,'The accuracy of this process is dependent where you start. The three-letter code, and\nthe resulting protein, will be completely different if translation starts on the first,\nsecond or third letter. For example, consider the sequence "AUGGCACGUUG". If translation\nstarts at the first base, the cell reads "AUG"s, "GCA" and "CGU"; and if you start at\nthe second base, the sequence would read "UGG", "CAC", and "GUU" . These options are\nreferred to as\xa0',(0,o.kt)("strong",{parentName:"p"},"reading frames"),", where the functional or\xa0",(0,o.kt)("strong",{parentName:"p"},"open"),'\xa0reading frame is one\nthat begins with the start codon "AUG". Changes to the sequence that starts, stops or\nmoves a reading frame dramatically alters the protein product. For example, the first\nread would produce met, ala and ala while the second would produce trp, his and val.\nMutations that change which blocks of three bases are used as codons are\ncalled\xa0',(0,o.kt)("strong",{parentName:"p"},"frameshift"),"\xa0mutations."),(0,o.kt)("p",null,(0,o.kt)("img",{alt:"Demonstration of the affect of reading frames on translation",src:n(637).Z,width:"2410",height:"1169"})))}h.isMDXComponent=!0},9690:function(e,t,n){t.Z=n.p+"assets/images/1.4-mRNA-to-Protein-db06a210206849655eb4d5a905613040.jpg"},9355:function(e,t,n){t.Z=n.p+"assets/images/1.5-codon-lookup-table-fd659602f31c30e3beb3f651c5de9368.jpg"},637:function(e,t,n){t.Z=n.p+"assets/images/1.6-reading-frames-e3af6973abae37e951cfd858747604a7.jpg"}}]);