"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[658],{3905:function(e,t,n){n.d(t,{Zo:function(){return u},kt:function(){return d}});var r=n(7294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function i(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function c(e,t){if(null==e)return{};var n,r,a=function(e,t){if(null==e)return{};var n,r,a={},o=Object.keys(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var s=r.createContext({}),m=function(e){var t=r.useContext(s),n=t;return e&&(n="function"==typeof e?e(t):i(i({},t),e)),n},u=function(e){var t=m(e.components);return r.createElement(s.Provider,{value:t},e.children)},l={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},p=r.forwardRef((function(e,t){var n=e.components,a=e.mdxType,o=e.originalType,s=e.parentName,u=c(e,["components","mdxType","originalType","parentName"]),p=m(n),d=a,f=p["".concat(s,".").concat(d)]||p[d]||l[d]||o;return n?r.createElement(f,i(i({ref:t},u),{},{components:n})):r.createElement(f,i({ref:t},u))}));function d(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var o=n.length,i=new Array(o);i[0]=p;var c={};for(var s in t)hasOwnProperty.call(t,s)&&(c[s]=t[s]);c.originalType=e,c.mdxType="string"==typeof e?e:a,i[1]=c;for(var m=2;m<o;m++)i[m]=n[m];return r.createElement.apply(null,i)}return r.createElement.apply(null,n)}p.displayName="MDXCreateElement"},2551:function(e,t,n){n.r(t),n.d(t,{assets:function(){return u},contentTitle:function(){return s},default:function(){return d},frontMatter:function(){return c},metadata:function(){return m},toc:function(){return l}});var r=n(7462),a=n(3366),o=(n(7294),n(3905)),i=["components"],c={title:"Copy Number Variation",track:"Genomic Variation"},s=void 0,m={unversionedId:"genomic-variation/copy-number-variation",id:"genomic-variation/copy-number-variation",title:"Copy Number Variation",description:"Copy number variants (CNVs) occur when parts of or entire chromosomes are",source:"@site/docs/02-genomic-variation/05-copy-number-variation.md",sourceDirName:"02-genomic-variation",slug:"/genomic-variation/copy-number-variation",permalink:"/docs/genomic-variation/copy-number-variation",draft:!1,editUrl:"https://github.com/stjude/learngenomics.dev/tree/main/docs/02-genomic-variation/05-copy-number-variation.md",tags:[],version:"current",sidebarPosition:5,frontMatter:{title:"Copy Number Variation",track:"Genomic Variation"},sidebar:"tutorialSidebar",previous:{title:"Structural Variation",permalink:"/docs/genomic-variation/structural-variation"},next:{title:"Variant Equivalence",permalink:"/docs/genomic-variation/variant-equivalence"}},u={},l=[],p={toc:l};function d(e){var t=e.components,c=(0,a.Z)(e,i);return(0,o.kt)("wrapper",(0,r.Z)({},p,c,{components:t,mdxType:"MDXLayout"}),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Copy number variants (CNVs)"),' occur when parts of or entire chromosomes are\nreplicated or lost. CNVs occur in normal healthy individuals, with no outward\nsign of disease. However, there are diseases that are due to CNVs such as Down\nSyndrome, where individuals have an extra copy of chromosome 21, or Huntington\'s\ndisease where a short sequence "CAG" is replicated over 35 times in a single\ngene. In cancer, cells with unstable genomes may duplicate sequences many times,\namplifying the genes that promote the disease, or cells may drop copies of genes\nthat prevent cancer.'),(0,o.kt)("p",null,(0,o.kt)("img",{alt:"Image depicting copy number variation and its effect on read mapping.",src:n(6699).Z,width:"2401",height:"1912"})),(0,o.kt)("p",null,"Detecting CNVs computationally from sequencing data begins by mapping reads to a\nreference genome and determining the read depth, or coverage. In theory, the\ncoverage should reflect the relative copy number. In practice, some DNA regions\nget more counts than others simply because their sequences are more easily\nmapped. In addition, duplicate reads generated by lab preparations may inflate\nlocal counts, and sequence machine errors give inflated quality estimates for\nsome bases. CNV detection methods adjust the coverage estimates to remove each\nof these effects. Coverage estimates at the base pair level are too small and\nvariable to reliably define a CNV, so smoothing methods are applied to derive a\nmean estimate of coverage within a window, or bin. Last, CNV methods that rely\non read depth identify the boundaries of CNVs regions through segmentation or\nmodeling methods."),(0,o.kt)("p",null,"In general, CNV detection methods further refine their search by comparing\nnormalized coverage estimates to a panel of normal control genomes with the\nexpected two chromosomal copies. In cancer, the mapped counts from a tumor\nsample are often compared to a normal genome from the same patient to further\nrefine the analysis. This comparison identifies changes due to disease rather\nthan the patient's natural variation."))}d.isMDXComponent=!0},6699:function(e,t,n){t.Z=n.p+"assets/images/2.4-Copy-Number-Variation-3d2aa2bdcede326ea79da34fe2de7eab.jpg"}}]);