"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[658],{3905:function(e,t,r){r.d(t,{Zo:function(){return l},kt:function(){return h}});var o=r(7294);function a(e,t,r){return t in e?Object.defineProperty(e,t,{value:r,enumerable:!0,configurable:!0,writable:!0}):e[t]=r,e}function n(e,t){var r=Object.keys(e);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);t&&(o=o.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),r.push.apply(r,o)}return r}function i(e){for(var t=1;t<arguments.length;t++){var r=null!=arguments[t]?arguments[t]:{};t%2?n(Object(r),!0).forEach((function(t){a(e,t,r[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(r)):n(Object(r)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(r,t))}))}return e}function c(e,t){if(null==e)return{};var r,o,a=function(e,t){if(null==e)return{};var r,o,a={},n=Object.keys(e);for(o=0;o<n.length;o++)r=n[o],t.indexOf(r)>=0||(a[r]=e[r]);return a}(e,t);if(Object.getOwnPropertySymbols){var n=Object.getOwnPropertySymbols(e);for(o=0;o<n.length;o++)r=n[o],t.indexOf(r)>=0||Object.prototype.propertyIsEnumerable.call(e,r)&&(a[r]=e[r])}return a}var s=o.createContext({}),m=function(e){var t=o.useContext(s),r=t;return e&&(r="function"==typeof e?e(t):i(i({},t),e)),r},l=function(e){var t=m(e.components);return o.createElement(s.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return o.createElement(o.Fragment,{},t)}},p=o.forwardRef((function(e,t){var r=e.components,a=e.mdxType,n=e.originalType,s=e.parentName,l=c(e,["components","mdxType","originalType","parentName"]),p=m(r),h=a,d=p["".concat(s,".").concat(h)]||p[h]||u[h]||n;return r?o.createElement(d,i(i({ref:t},l),{},{components:r})):o.createElement(d,i({ref:t},l))}));function h(e,t){var r=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var n=r.length,i=new Array(n);i[0]=p;var c={};for(var s in t)hasOwnProperty.call(t,s)&&(c[s]=t[s]);c.originalType=e,c.mdxType="string"==typeof e?e:a,i[1]=c;for(var m=2;m<n;m++)i[m]=r[m];return o.createElement.apply(null,i)}return o.createElement.apply(null,r)}p.displayName="MDXCreateElement"},2551:function(e,t,r){r.r(t),r.d(t,{assets:function(){return l},contentTitle:function(){return s},default:function(){return h},frontMatter:function(){return c},metadata:function(){return m},toc:function(){return u}});var o=r(7462),a=r(3366),n=(r(7294),r(3905)),i=["components"],c={title:"Copy Number Variation",track:"Genomic Variation"},s=void 0,m={unversionedId:"genomic-variation/copy-number-variation",id:"genomic-variation/copy-number-variation",title:"Copy Number Variation",description:"Revisiting our bakery analogy, suppose a baker wants to produce more chocolate cakes, but not all of the same kind. Our baker may buy multiple copies of a chocolate cake cookbook or copy the chocolate cake chapter from their favorite book a few times. Alternatively, the baker may decide they will no longer make coconut cakes and cut that chapter from the book to prevent the staff from baking them. These kinds of changes are analogous to copy number variants (CNVs) where chromosomes or parts of chromosomes are replicated or lost.",source:"@site/docs/02-genomic-variation/05-copy-number-variation.md",sourceDirName:"02-genomic-variation",slug:"/genomic-variation/copy-number-variation",permalink:"/docs/genomic-variation/copy-number-variation",editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/02-genomic-variation/05-copy-number-variation.md",tags:[],version:"current",sidebarPosition:5,frontMatter:{title:"Copy Number Variation",track:"Genomic Variation"},sidebar:"tutorialSidebar",previous:{title:"Structural Variation",permalink:"/docs/genomic-variation/structural-variation"},next:{title:"Summary",permalink:"/docs/genomic-variation/summary"}},l={},u=[],p={toc:u};function h(e){var t=e.components,c=(0,a.Z)(e,i);return(0,n.kt)("wrapper",(0,o.Z)({},p,c,{components:t,mdxType:"MDXLayout"}),(0,n.kt)("p",null,"Revisiting our bakery analogy, suppose a baker wants to produce more chocolate cakes, but not all of the same kind. Our baker may buy multiple copies of a chocolate cake cookbook or copy the chocolate cake chapter from their favorite book a few times. Alternatively, the baker may decide they will no longer make coconut cakes and cut that chapter from the book to prevent the staff from baking them. These kinds of changes are analogous to ",(0,n.kt)("strong",{parentName:"p"},"copy number variants (CNVs)")," where chromosomes or parts of chromosomes are replicated or lost."),(0,n.kt)("p",null,'CNVs occur in normal healthy individuals, with no outward sign of disease. However, there are diseases that are due to CNVs such as Down Syndrome, where individuals have an extra copy of chromosome 21, or Huntington\\\'s disease where a short sequence "CAG" is replicated over 35 times in a single gene. In cancer, cells with unstable genomes may duplicate sequences many times, amplifying the genes that promote the disease, or cells may drop copies of genes that prevent cancer.'),(0,n.kt)("p",null,"Detecting CNVs computationally from sequencing data begins by mapping reads to a reference genome and determining the read depth, or coverage. In theory, the coverage should reflect the copy number. In practice, some DNA regions get more counts than others simply because their sequences are more easily mapped. In addition, duplicate reads generated by lab preparations may inflate local counts, and sequence machine errors give inflated quality estimates for some bases. CNV detection methods adjust the coverage estimates to remove each of these effects. Coverage estimates at the base pair level are too small and variable to reliably define a CNV, so smoothing methods are applied to derive a mean estimate of coverage within a window, or bin. Last, CNV methods that rely on read depth identify the boundaries of CNVs regions through segmentation or modeling methods."),(0,n.kt)("p",null,"In general, CNV detection methods further refine their search by comparing normalized coverage estimates to a panel of normal control genomes with the expected two chromosomal copies. In cancer, the mapped counts from a tumor sample are often compared to a normal genome from the same patient to further refine the analysis. This comparison identifies changes due to disease rather than the patient\\'s natural variation."),(0,n.kt)("p",null,(0,n.kt)("img",{src:r(5382).Z,width:"2000",height:"1593"})))}h.isMDXComponent=!0},5382:function(e,t,r){t.Z=r.p+"assets/images/3.6-CNV-1b1d9c02610f48517b8f5ef6ea117531.jpg"}}]);