"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[691],{3905:function(e,n,t){t.d(n,{Zo:function(){return l},kt:function(){return m}});var r=t(7294);function i(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function o(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);n&&(r=r.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,r)}return t}function a(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?o(Object(t),!0).forEach((function(n){i(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):o(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function c(e,n){if(null==e)return{};var t,r,i=function(e,n){if(null==e)return{};var t,r,i={},o=Object.keys(e);for(r=0;r<o.length;r++)t=o[r],n.indexOf(t)>=0||(i[t]=e[t]);return i}(e,n);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(r=0;r<o.length;r++)t=o[r],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(i[t]=e[t])}return i}var s=r.createContext({}),u=function(e){var n=r.useContext(s),t=n;return e&&(t="function"==typeof e?e(n):a(a({},n),e)),t},l=function(e){var n=u(e.components);return r.createElement(s.Provider,{value:n},e.children)},p={inlineCode:"code",wrapper:function(e){var n=e.children;return r.createElement(r.Fragment,{},n)}},d=r.forwardRef((function(e,n){var t=e.components,i=e.mdxType,o=e.originalType,s=e.parentName,l=c(e,["components","mdxType","originalType","parentName"]),d=u(t),m=i,f=d["".concat(s,".").concat(m)]||d[m]||p[m]||o;return t?r.createElement(f,a(a({ref:n},l),{},{components:t})):r.createElement(f,a({ref:n},l))}));function m(e,n){var t=arguments,i=n&&n.mdxType;if("string"==typeof e||i){var o=t.length,a=new Array(o);a[0]=d;var c={};for(var s in n)hasOwnProperty.call(n,s)&&(c[s]=n[s]);c.originalType=e,c.mdxType="string"==typeof e?e:i,a[1]=c;for(var u=2;u<o;u++)a[u]=t[u];return r.createElement.apply(null,a)}return r.createElement.apply(null,t)}d.displayName="MDXCreateElement"},8222:function(e,n,t){t.r(n),t.d(n,{assets:function(){return l},contentTitle:function(){return s},default:function(){return m},frontMatter:function(){return c},metadata:function(){return u},toc:function(){return p}});var r=t(7462),i=t(3366),o=(t(7294),t(3905)),a=["components"],c={title:"Next-Generation Sequencing",track:"Next-Generation Sequencing",index:!0},s=void 0,u={unversionedId:"next-generation-sequencing/index",id:"next-generation-sequencing/index",title:"Next-Generation Sequencing",description:"This chapter discusses the process of how tissue goes from physical material to a file containing the genomic/transcriptomic sequence which can be processed by a computer. This section minimizes the chemical details and focus only on the essentials required to work with data. Understanding how genomic data is acquired helps explain how analyses are done. If you are interested in a more complete (although not exhaustive) description of these processes, see the appendix guide at the end of this section.",source:"@site/docs/03-next-generation-sequencing/index.md",sourceDirName:"03-next-generation-sequencing",slug:"/next-generation-sequencing/",permalink:"learngenomics.dev/docs/next-generation-sequencing/",draft:!1,editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/03-next-generation-sequencing/index.md",tags:[],version:"current",frontMatter:{title:"Next-Generation Sequencing",track:"Next-Generation Sequencing",index:!0},sidebar:"tutorialSidebar",previous:{title:"Summary",permalink:"learngenomics.dev/docs/genomic-variation/summary"},next:{title:"Sample Collection",permalink:"learngenomics.dev/docs/next-generation-sequencing/sample-collection"}},l={},p=[{value:"Untriaged",id:"untriaged",level:2}],d={toc:p};function m(e){var n=e.components,t=(0,i.Z)(e,a);return(0,o.kt)("wrapper",(0,r.Z)({},d,t,{components:n,mdxType:"MDXLayout"}),(0,o.kt)("p",null,"This chapter discusses the process of how tissue goes from physical material to a file containing the genomic/transcriptomic sequence which can be processed by a computer. This section minimizes the chemical details and focus only on the essentials required to work with data. Understanding how genomic data is acquired helps explain how analyses are done. If you are interested in a more complete (although not exhaustive) description of these processes, see the appendix guide at the end of this section."),(0,o.kt)("p",null,"All of the details below are in reference to paired-end reads? (linked reads from both ends of the same sequence fragment), short read sequencing (less than 1000 bp) using the latest best practices on the Illumina Platform."),(0,o.kt)("h2",{id:"untriaged"},"Untriaged"),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"error sources due to Illumina technology")),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"how read pairing is used")),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"a description of what a flow cell is"))),(0,o.kt)("p",null,"Gene models ","-","- ENSEMBL, UCSC, NCBI"))}m.isMDXComponent=!0}}]);