"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[294],{3905:function(e,n,t){t.d(n,{Zo:function(){return u},kt:function(){return f}});var i=t(7294);function o(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function r(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);n&&(i=i.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,i)}return t}function a(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?r(Object(t),!0).forEach((function(n){o(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):r(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function s(e,n){if(null==e)return{};var t,i,o=function(e,n){if(null==e)return{};var t,i,o={},r=Object.keys(e);for(i=0;i<r.length;i++)t=r[i],n.indexOf(t)>=0||(o[t]=e[t]);return o}(e,n);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(i=0;i<r.length;i++)t=r[i],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(o[t]=e[t])}return o}var c=i.createContext({}),l=function(e){var n=i.useContext(c),t=n;return e&&(t="function"==typeof e?e(n):a(a({},n),e)),t},u=function(e){var n=l(e.components);return i.createElement(c.Provider,{value:n},e.children)},p={inlineCode:"code",wrapper:function(e){var n=e.children;return i.createElement(i.Fragment,{},n)}},d=i.forwardRef((function(e,n){var t=e.components,o=e.mdxType,r=e.originalType,c=e.parentName,u=s(e,["components","mdxType","originalType","parentName"]),d=l(t),f=o,g=d["".concat(c,".").concat(f)]||d[f]||p[f]||r;return t?i.createElement(g,a(a({ref:n},u),{},{components:t})):i.createElement(g,a({ref:n},u))}));function f(e,n){var t=arguments,o=n&&n.mdxType;if("string"==typeof e||o){var r=t.length,a=new Array(r);a[0]=d;var s={};for(var c in n)hasOwnProperty.call(n,c)&&(s[c]=n[c]);s.originalType=e,s.mdxType="string"==typeof e?e:o,a[1]=s;for(var l=2;l<r;l++)a[l]=t[l];return i.createElement.apply(null,a)}return i.createElement.apply(null,t)}d.displayName="MDXCreateElement"},821:function(e,n,t){t.r(n),t.d(n,{assets:function(){return u},contentTitle:function(){return c},default:function(){return f},frontMatter:function(){return s},metadata:function(){return l},toc:function(){return p}});var i=t(7462),o=t(3366),r=(t(7294),t(3905)),a=["components"],s={title:"Genomic Regions and Transcription",track:"Biological Foundations"},c=void 0,l={unversionedId:"biological-foundations/genomic-regions-and-transcription",id:"biological-foundations/genomic-regions-and-transcription",title:"Genomic Regions and Transcription",description:'Given that a primary function of DNA is to guide the production of proteins, it might be assumed that genomic DNA is dominated by the sequences of genes that encode proteins. Somewhat counterintuitively, this is not the case. In fact, it is estimated that\xa0only 1% of the human genome is part of a "coding region", meaning the region contains information used to create proteins. The function of the other 99%, known as "non-coding regions" of the genome, is an area of active research.',source:"@site/docs/01-biological-foundations/03-genomic-regions-and-transcription.md",sourceDirName:"01-biological-foundations",slug:"/biological-foundations/genomic-regions-and-transcription",permalink:"/docs/biological-foundations/genomic-regions-and-transcription",editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/01-biological-foundations/03-genomic-regions-and-transcription.md",tags:[],version:"current",sidebarPosition:3,frontMatter:{title:"Genomic Regions and Transcription",track:"Biological Foundations"},sidebar:"tutorialSidebar",previous:{title:"The Central Dogma of Molecular Biology",permalink:"/docs/biological-foundations/the-central-dogma"},next:{title:"Translation: Decoding the Transcript",permalink:"/docs/biological-foundations/decoding-the-transcript"}},u={},p=[],d={toc:p};function f(e){var n=e.components,s=(0,o.Z)(e,a);return(0,r.kt)("wrapper",(0,i.Z)({},d,s,{components:n,mdxType:"MDXLayout"}),(0,r.kt)("p",null,"Given that a primary function of DNA is to guide the production of proteins, it might be assumed that genomic DNA is dominated by the sequences of genes that encode proteins. Somewhat counterintuitively, this is not the case. In fact, it is estimated that\xa0",(0,r.kt)("a",{parentName:"p",href:"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439153/"},'only 1% of the human genome is part of a "coding region"'),', meaning the region contains information used to create proteins. The function of the other 99%, known as "non-coding regions" of the genome, is an area of active research.'),(0,r.kt)("p",null,"When a gene is transcribed from DNA to RNA through the process of transcription, a\xa0",(0,r.kt)("strong",{parentName:"p"},"mature mRNA"),'\xa0(or "mature messenger RNA", sometimes just shortened to "mRNA" where the "mature" part is implied) is typically the desired final product. During the first step of transcription, the relevant coding region of a gene within the DNA sequence is turned into a single-stranded, immature version of RNA. This immature RNA is not used directly to encode for proteins. Instead, it is cut up into pieces and ',(0,r.kt)("strong",{parentName:"p"},"spliced")," together to produce the working recipe as mature mRNA. The sections of the gene which are included in the final recipe are called the\xa0",(0,r.kt)("strong",{parentName:"p"},"exons"),", while the sections of the sequence that are spliced out of the recipe are called\xa0",(0,r.kt)("strong",{parentName:"p"},"introns"),". The figure below depicts how introns and exons are related to the final mRNA recipe molecule."),(0,r.kt)("p",null,(0,r.kt)("img",{alt:"Figure showing DNA being transcribed into immature RNA and then eventually spliced to mature messenger RNA.",src:t(6584).Z,width:"2579",height:"1635"})),(0,r.kt)("p",null,'To use the analogy presented in the last chapter, each recipe (gene) has a number of variations or "flavors" that can be added to tweak the final cake that is made. At one point in time, the cell may make a version of the gene above by including all four exons above. At some later point in time, the cell might make a version of this protein only including exons 1, 3, 4, which would have a slightly different affect. These are called ',(0,r.kt)("strong",{parentName:"p"},"alternative splicing")," events and are an area of active research."),(0,r.kt)("p",null,"Variations in the exonic regions of the genome can significantly change which parts are used to assemble a protein, thereby, changing its function in a cell, like changing sugar to salt in a recipe. Variations in non-coding regions outside exonic sequencing are still important (examples of important non-coding genomic sequences include\xa0",(0,r.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Transcription_factor"},"transcription factor binding sites"),"\xa0and\xa0",(0,r.kt)("a",{parentName:"p",href:"https://en.wikipedia.org/wiki/Super-enhancer"},"super-enhancers"),'). The effect of variations in non-coding DNA are not as easily interpreted as exonic variants, but frequently they are significant in disease and are discussed in the\xa0"Genomic Variation"\xa0section of this book.'))}f.isMDXComponent=!0},6584:function(e,n,t){n.Z=t.p+"assets/images/1.3-DNA-to-mRNA-e7abc7a704fb717844a97663b255ae26.jpg"}}]);