"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[7987],{3905:function(e,t,n){n.d(t,{Zo:function(){return d},kt:function(){return h}});var a=n(7294);function i(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);t&&(a=a.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,a)}return n}function o(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){i(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,a,i=function(e,t){if(null==e)return{};var n,a,i={},r=Object.keys(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||(i[n]=e[n]);return i}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(a=0;a<r.length;a++)n=r[a],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(i[n]=e[n])}return i}var l=a.createContext({}),c=function(e){var t=a.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):o(o({},t),e)),n},d=function(e){var t=c(e.components);return a.createElement(l.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return a.createElement(a.Fragment,{},t)}},p=a.forwardRef((function(e,t){var n=e.components,i=e.mdxType,r=e.originalType,l=e.parentName,d=s(e,["components","mdxType","originalType","parentName"]),p=c(n),h=i,g=p["".concat(l,".").concat(h)]||p[h]||u[h]||r;return n?a.createElement(g,o(o({ref:t},d),{},{components:n})):a.createElement(g,o({ref:t},d))}));function h(e,t){var n=arguments,i=t&&t.mdxType;if("string"==typeof e||i){var r=n.length,o=new Array(r);o[0]=p;var s={};for(var l in t)hasOwnProperty.call(t,l)&&(s[l]=t[l]);s.originalType=e,s.mdxType="string"==typeof e?e:i,o[1]=s;for(var c=2;c<r;c++)o[c]=n[c];return a.createElement.apply(null,o)}return a.createElement.apply(null,n)}p.displayName="MDXCreateElement"},4890:function(e,t,n){n.r(t),n.d(t,{assets:function(){return d},contentTitle:function(){return l},default:function(){return h},frontMatter:function(){return s},metadata:function(){return c},toc:function(){return u}});var a=n(7462),i=n(3366),r=(n(7294),n(3905)),o=["components"],s={title:"Next-Generation Sequencing",track:"Next-Generation Sequencing"},l=void 0,c={unversionedId:"next-generation-sequencing/next-generation-sequencing",id:"next-generation-sequencing/next-generation-sequencing",title:"Next-Generation Sequencing",description:'Once collected, sequencing the sample begins. The sample is removed from storage, and a portion is taken to prepare what is called a "library" for sequencing. This is always performed in a wet lab environment, and the chemical methods can be quite complex.',source:"@site/docs/03-next-generation-sequencing/02-next-generation-sequencing.md",sourceDirName:"03-next-generation-sequencing",slug:"/next-generation-sequencing/next-generation-sequencing",permalink:"/docs/next-generation-sequencing/next-generation-sequencing",editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/03-next-generation-sequencing/02-next-generation-sequencing.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{title:"Next-Generation Sequencing",track:"Next-Generation Sequencing"},sidebar:"tutorialSidebar",previous:{title:"Sample Collection",permalink:"/docs/next-generation-sequencing/sample-collection"},next:{title:"Read Mapping",permalink:"/docs/next-generation-sequencing/read-mapping"}},d={},u=[{value:"Types of Sequencing",id:"types-of-sequencing",level:2},{value:"Common Sequence Applications",id:"common-sequence-applications",level:2},{value:"Whole Genome Sequencing",id:"whole-genome-sequencing",level:2},{value:"Whole Exome Sequencing",id:"whole-exome-sequencing",level:2},{value:"Whole Transcriptome Sequencing / RNA-seq",id:"whole-transcriptome-sequencing--rna-seq",level:2},{value:"Targeted Sequencing",id:"targeted-sequencing",level:2}],p={toc:u};function h(e){var t=e.components,s=(0,i.Z)(e,o);return(0,r.kt)("wrapper",(0,a.Z)({},p,s,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("p",null,'Once collected, sequencing the sample begins. The sample is removed from storage, and a portion is taken to prepare what is called a "library" for sequencing. This is always performed in a wet lab environment, and the chemical methods can be quite complex.'),(0,r.kt)("p",null,"While it sounds easy to read the entire genome by laying all of the chromosomes end-to-end, in practice, there are some problems with this approach. Chiefly, researchers currently lack the biochemical processing abilities to extract to read such long stretches of DNA quickly and accurately. Additionally, next generation sequencing technologies have their own set of limitations, including:"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Each nucleotide the sequencer calls is accompanied by a confidence score estimating the how likely the recorded call is true. In short, the sequencer\\'s calls are not perfect.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"As the sequencer observes each nucleotide along the read, the confidence it has in each new call drops so that if allowed to continue eventually the data quality of the last bases would be too low to use reliably.")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Sequencers can only read DNA fragments because RNA is too unstable. So biochemical methods are employed to reverse the transcription process and produce a DNA fragment that captures the information in the RNA sequence."))),(0,r.kt)("p",null,"Given these physical limitations, the process, called ",(0,r.kt)("strong",{parentName:"p"},"shotgun sequencing"),", looks more like the following:"),(0,r.kt)("ol",null,(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},"The sample's DNA is fragmented into billions of smaller DNA pieces that are somewhere between tens to thousands of bases long.")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},'This soup of DNA fragments with randomly distributed lengths is then put through a filtering step called "size selection". Only DNA fragments of a certain length will remain (typically somewhere between 200-500 bp, where bp stands for base pair).')),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},'Among many other important steps to prepare the DNA library for sequencing is "adapter ligation". This process involves tacking on special sequences to both ends of the DNA fragment which is required by the sequencing process.'))),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(7924).Z,width:"1000",height:"352"})),(0,r.kt)("ol",{start:4},(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},"Optionally, a chemical amplification step can be conducted to increase the amount of genomic material available for sequencing.")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},"The sample is loaded onto a flow cell and put into an Illumina sequencer, which is capable of reading millions of DNA fragments and capturing the results digitally.")),(0,r.kt)("li",{parentName:"ol"},(0,r.kt)("p",{parentName:"li"},'Once the sequencing is complete, the sequencer closes the sequencing session by creating a directory of BCL files (Illumina\\\'s proprietary storage format). These are immediately converted to a FASTQ file (standard DNA storage format), typically the "input" to the computational team.'))),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(5262).Z,width:"1000",height:"745"})),(0,r.kt)("p",null,'For our paired-end sequencing example, it is important to note how the DNA sequencer reads the data in step 4. In each 200-500bp DNA fragment, the sequencer begins reading from one of the fragments and drops off somewhere in the middle. It stores all of the nucleotides read from this end of the fragment with their associated confidence scores as "Read 1". After this is complete, the sequencer reads from the opposite end of the fragment and stores the nucleotide calls and confidence scores known as "Read 2". Together, the combination of a Read 1 from a DNA fragment and a Read 2 from the other end of the DNA fragment is called a "read pair" as denoted in Figure ',"#","."),(0,r.kt)("p",null,'After the steps above have completed, the result is a single FASTQ file (single-end sequencing) or a pair of FASTQ files (paired-end). For more information on the FASTQ file format, see its section in the "Genomic File Formats" chapter.'),(0,r.kt)("h2",{id:"types-of-sequencing"},"Types of Sequencing"),(0,r.kt)("p",null,'Sequencing is laboratory technique that identifies and determines the order of bases in a DNA molecule. Each sequencing type is best analyzed by a different computational method and is interpreted in different way. The scientific goals of the research, cost and practical limits determine which types of sequencing are chosen. For example, clinical sequencing seeks the time-sensitive detection of known mutations, while population studies focus on the discovery of new variants. Often, the type of sequencing ordered is a trade-off between cost and the breadth of information. This section discusses what scientists weigh when deciding which pieces of the genome/exome/transcriptome needs sequencing; the "Data Acquisition" section of this guide focuses on how\xa0the genome/exome/transcriptome are sequenced.'),(0,r.kt)("h2",{id:"common-sequence-applications"},"Common Sequence Applications"),(0,r.kt)("table",null,(0,r.kt)("thead",{parentName:"table"},(0,r.kt)("tr",{parentName:"thead"},(0,r.kt)("th",{parentName:"tr",align:null},(0,r.kt)("strong",{parentName:"th"},"Area of the genome")),(0,r.kt)("th",{parentName:"tr",align:null},(0,r.kt)("strong",{parentName:"th"},"Abbreviations")),(0,r.kt)("th",{parentName:"tr",align:null},(0,r.kt)("strong",{parentName:"th"},"Description")))),(0,r.kt)("tbody",{parentName:"table"},(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Whole Genome Sequencing"),(0,r.kt)("td",{parentName:"tr",align:null},"WGS"),(0,r.kt)("td",{parentName:"tr",align:null},"Characterizes the entire genome reading sequences from DNA")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Whole Exome Sequencing"),(0,r.kt)("td",{parentName:"tr",align:null},"WES, WXS"),(0,r.kt)("td",{parentName:"tr",align:null},"Characterizes the exonic, or coding, regions of the genome reading sequences from DNA")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Transcriptome Sequencing"),(0,r.kt)("td",{parentName:"tr",align:null},"WTS, RNA-seq"),(0,r.kt)("td",{parentName:"tr",align:null},"Characterizes the exonic regions that are actively expressed in the cell reading sequences from RNA")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Targeted Sequencing"),(0,r.kt)("td",{parentName:"tr",align:null},"Validation Capture, Valcap"),(0,r.kt)("td",{parentName:"tr",align:null},"Characterizes a chosen region from DNA, can be a coding or non-coding region. The region can be coding or non-coding and is typically focused, high-depth sequencing for validating variants")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"ChIP seq"),(0,r.kt)("td",{parentName:"tr",align:null},"ChIP-seq"),(0,r.kt)("td",{parentName:"tr",align:null},"Characterizes any sequence the genomic locations where a protein interacts with DNA, both coding and non-coding")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"ATAC sequencing"),(0,r.kt)("td",{parentName:"tr",align:null},"ATAC-seq"),(0,r.kt)("td",{parentName:"tr",align:null},'Characterizes any sequence where DNA does not interact with proteins or is "open", both coding and non-coding')),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Hi-C"),(0,r.kt)("td",{parentName:"tr",align:null},"Hi-C"),(0,r.kt)("td",{parentName:"tr",align:null},"Characterizes the genomic locations where chromosomes come in close contact with each other. Typically used in characterizing 3D interactions of DNA")),(0,r.kt)("tr",{parentName:"tbody"},(0,r.kt)("td",{parentName:"tr",align:null},"Single Cell Sequencing"),(0,r.kt)("td",{parentName:"tr",align:null},"scDNA-seq, scRNA-seq"),(0,r.kt)("td",{parentName:"tr",align:null},"DNA or RNA sequencing of individual cells, captures granularity at an individual cell level rather than a mixture of cells, can be coding or non-coding with lower coverage")))),(0,r.kt)("h2",{id:"whole-genome-sequencing"},"Whole Genome Sequencing"),(0,r.kt)("p",null,"If cost was not an issue, whole genome sequencing (",(0,r.kt)("strong",{parentName:"p"},"WGS"),") would be the standard approach for detecting variants. The advantage of WGS is a complete survey of the DNA, allowing the researcher to detect all the genomic variation for a given sample. WGS is needed when exploring how variations in non-coding regions influence the cells. Further, WGS is needed to detect the scope of variants that affect large regions of the chromosome (Structural Variation and Copy Number Variation, which are discussed in Chapter 3). The disadvantage, other than cost, is that many of variations are not relevant to disease or the research question and need to be carefully filtered out. WGS is used in basic exploratory research and increasingly in the clinic as costs drop."),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(9752).Z,width:"2000",height:"607"})),(0,r.kt)("h2",{id:"whole-exome-sequencing"},"Whole Exome Sequencing"),(0,r.kt)("p",null,"Whole exome sequencing (",(0,r.kt)("strong",{parentName:"p"},"WES"),") captures the coding sequences of known genes from DNA and is a focused, cost-effective approach to characterizing what protein byproducts might be made. WES is commonly used in the clinic to rapidly detect mutations in tumor samples so that treatments can be tailored to the individual. It is less commonly used in basic research due to its limited scope. The exceptions are large scale population studies with thousands of samples where whole genome approaches are too expensive."),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(7008).Z,width:"1000",height:"326"})),(0,r.kt)("h2",{id:"whole-transcriptome-sequencing--rna-seq"},"Whole Transcriptome Sequencing / RNA-seq"),(0,r.kt)("p",null,"Whole transcriptome sequencing, or\xa0",(0,r.kt)("strong",{parentName:"p"},"RNA-seq"),", captures the working recipes of genes as they float around a cell. This sequencing type gives the researcher a window into the mixture of proteins being created by the cell when the sample was collected. Mutations and natural variations, especially those which alter protein byproducts, can be detected using RNA-seq. Additionally, a common goal of RNA-seq is to quantify the transcripts of active genes to understand which genes are turned on and off in response to experimental treatments, disease or developmental changes in the cell."),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(4374).Z,width:"2000",height:"651"})),(0,r.kt)("h2",{id:"targeted-sequencing"},"Targeted Sequencing"),(0,r.kt)("p",null,(0,r.kt)("strong",{parentName:"p"},"Targeted sequencing")," or validation capture sequencing (",(0,r.kt)("strong",{parentName:"p"},"Valcap"),") captures preselected regions in the genome based on experimental design. Often targeted sequencing is used to capture a subset of important genes (called ",(0,r.kt)("strong",{parentName:"p"},"gene panels"),") in standardized tests when cost and time are limited Alternatively, targeted sequencing is frequently used to gain greater insight into a region observed through a whole genome or whole exome study where there is less read depth. The method can be tailored for coding or non-coding variants."),(0,r.kt)("p",null,(0,r.kt)("img",{src:n(2064).Z,width:"1000",height:"564"})))}h.isMDXComponent=!0},9752:function(e,t,n){t.Z=n.p+"assets/images/2.1-WGS-d88352e5c737eca2fb4974e54420a7c2.jpg"},7008:function(e,t,n){t.Z=n.p+"assets/images/2.2-WES-d54690a3e4a2bcdf075c8481e4ca6f5d.jpg"},4374:function(e,t,n){t.Z=n.p+"assets/images/2.3-RNASeq-2a757ea6fd59281d9e647bf88de14d33.jpg"},2064:function(e,t,n){t.Z=n.p+"assets/images/2.4-Targeted-d15a6add663d900051894629579077e9.jpg"},7924:function(e,t,n){t.Z=n.p+"assets/images/4.1-Read-on-Flow-Cell-Zoomed-ed0ea55a052a14fd30a44bbd97cae2d3.jpg"},5262:function(e,t,n){t.Z=n.p+"assets/images/4.2-Reads-on-Flow-Cell-c72528a8a3f89d7afe95115bac0b5ae6.jpg"}}]);