"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[1400],{3905:function(e,t,n){n.d(t,{Zo:function(){return d},kt:function(){return h}});var r=n(7294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function i(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function s(e,t){if(null==e)return{};var n,r,a=function(e,t){if(null==e)return{};var n,r,a={},o=Object.keys(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var c=r.createContext({}),l=function(e){var t=r.useContext(c),n=t;return e&&(n="function"==typeof e?e(t):i(i({},t),e)),n},d=function(e){var t=l(e.components);return r.createElement(c.Provider,{value:t},e.children)},p={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},u=r.forwardRef((function(e,t){var n=e.components,a=e.mdxType,o=e.originalType,c=e.parentName,d=s(e,["components","mdxType","originalType","parentName"]),u=l(n),h=a,m=u["".concat(c,".").concat(h)]||u[h]||p[h]||o;return n?r.createElement(m,i(i({ref:t},d),{},{components:n})):r.createElement(m,i({ref:t},d))}));function h(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var o=n.length,i=new Array(o);i[0]=u;var s={};for(var c in t)hasOwnProperty.call(t,c)&&(s[c]=t[c]);s.originalType=e,s.mdxType="string"==typeof e?e:a,i[1]=s;for(var l=2;l<o;l++)i[l]=n[l];return r.createElement.apply(null,i)}return r.createElement.apply(null,n)}u.displayName="MDXCreateElement"},8436:function(e,t,n){n.r(t),n.d(t,{assets:function(){return d},contentTitle:function(){return c},default:function(){return h},frontMatter:function(){return s},metadata:function(){return l},toc:function(){return p}});var r=n(7462),a=n(3366),o=(n(7294),n(3905)),i=["components"],s={title:"Translation: Decoding the Transcript",track:"Biological Foundations"},c=void 0,l={unversionedId:"biological-foundations/decoding-the-transcript",id:"biological-foundations/decoding-the-transcript",title:"Translation: Decoding the Transcript",description:"After DNA is transcribed to RNA and the RNA is spliced, the next step in protein production is translation. The genetic code that cells use to translate RNA to protein is universally preserved across organisms. So you, your dog, and the lettuce in your salad all read and translate the genetic code the same. This helps scientists determine which changes in a gene have functional consequences by comparing transcripts from different species. Further, the universality of this code makes modeling genetic diseases in an animal like mice informative.",source:"@site/docs/01-biological-foundations/04-decoding-the-transcript.md",sourceDirName:"01-biological-foundations",slug:"/biological-foundations/decoding-the-transcript",permalink:"/docs/biological-foundations/decoding-the-transcript",editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/01-biological-foundations/04-decoding-the-transcript.md",tags:[],version:"current",sidebarPosition:4,frontMatter:{title:"Translation: Decoding the Transcript",track:"Biological Foundations"},sidebar:"tutorialSidebar",previous:{title:"Genomic Regions and Transcription",permalink:"/docs/biological-foundations/genomic-regions-and-transcription"},next:{title:"Genotype, Phenotype and Haplotype",permalink:"/docs/biological-foundations/genotype-phenotype-haplotype"}},d={},p=[{value:"Reading Frames",id:"reading-frames",level:2}],u={toc:p};function h(e){var t=e.components,s=(0,a.Z)(e,i);return(0,o.kt)("wrapper",(0,r.Z)({},u,s,{components:t,mdxType:"MDXLayout"}),(0,o.kt)("p",null,"After DNA is transcribed to RNA and the RNA is spliced, the next step in protein production is translation. The genetic code that cells use to translate RNA to protein is universally preserved across organisms. So you, your dog, and the lettuce in your salad all read and translate the genetic code the same. This helps scientists determine which changes in a gene have functional consequences by comparing transcripts from different species. Further, the universality of this code makes modeling genetic diseases in an animal like mice informative."),(0,o.kt)("p",null,"The mRNA transcript is encoded in three-letter blocks known as\xa0",(0,o.kt)("strong",{parentName:"p"},"codons"),". Codons specify that a particular protein building block, or ",(0,o.kt)("strong",{parentName:"p"},"amino acid"),', should be added to a growing protein chain. For example, translating from mRNA, "CAG" specifies to add a glutamine. As shown in Figure 1.4, other codons specify that translation should start or end at a particular spot. Because multiple codons stand for the same amino acid, some substitutions may not change which amino acid is present. This type of variation, where the mRNA sequence change but the amino acid sequence stays the same, are called "silent mutations." When codon variants change the amino acid, the effect ranges from insignificant to deleterious. Computational biologists use programs that incorporate knowledge of the codons and chemistry of proteins to predict the effect of a given variant.'),(0,o.kt)("p",null,(0,o.kt)("img",{src:n(5990).Z,width:"1178",height:"1069"})),(0,o.kt)("p",null,"As a short aside, DNA and RNA use different coding alphabets. That is, in a mRNA transcript, the DNA base thymine (T) is substituted by uracil (U)."),(0,o.kt)("table",null,(0,o.kt)("thead",{parentName:"table"},(0,o.kt)("tr",{parentName:"thead"},(0,o.kt)("th",{parentName:"tr",align:null},(0,o.kt)("strong",{parentName:"th"},"Molecule")),(0,o.kt)("th",{parentName:"tr",align:null},(0,o.kt)("strong",{parentName:"th"},"Alphabet")))),(0,o.kt)("tbody",{parentName:"table"},(0,o.kt)("tr",{parentName:"tbody"},(0,o.kt)("td",{parentName:"tr",align:null},"DNA"),(0,o.kt)("td",{parentName:"tr",align:null},"ACGT")),(0,o.kt)("tr",{parentName:"tbody"},(0,o.kt)("td",{parentName:"tr",align:null},"RNA"),(0,o.kt)("td",{parentName:"tr",align:null},"ACGU (Uracil replaces Thymine, i.e., U replaces T)")),(0,o.kt)("tr",{parentName:"tbody"},(0,o.kt)("td",{parentName:"tr",align:null},"Proteins"),(0,o.kt)("td",{parentName:"tr",align:null},"Uses a series of triplets in RNA (codons) to encode an amino acid sequence")))),(0,o.kt)("h2",{id:"reading-frames"},"Reading Frames"),(0,o.kt)("p",null,'The accuracy of this process is dependent where you start. The three-letter code, and the resulting protein, will be completely different if translation starts on the first, second or third letter. For example, consider the sequence "AUGGCACGUUG". If translation starts at the first base, the cell reads "AUG"s, "GCA" and "CGU"; and if you start at the second base, the sequence would read "UGG", "CAC", and "GUU" . These options are referred to as\xa0',(0,o.kt)("strong",{parentName:"p"},"reading frames"),", where the functional or\xa0",(0,o.kt)("strong",{parentName:"p"},"open"),'\xa0reading frame is one that begins with the start codon "AUG". Changes to the sequence that starts, stops or moves a reading frame dramatically alters the protein product. For example, the first read would produce met, ala and ala while the second would produce trp, his and val. Mutations that change which blocks of three bases are used as codons are called\xa0',(0,o.kt)("strong",{parentName:"p"},"frameshift"),"\xa0mutations."))}h.isMDXComponent=!0},5990:function(e,t,n){t.Z=n.p+"assets/images/1.4-Codons-fd659602f31c30e3beb3f651c5de9368.jpg"}}]);