"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[53],{1109:function(e){e.exports=JSON.parse('{"pluginId":"default","version":"current","label":"Next","banner":null,"badge":false,"className":"docs-version-current","isLast":true,"docsSidebars":{"tutorialSidebar":[{"type":"category","label":"Biological Foundations","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Cells, Chromosomes, and Genomes","href":"/docs/biological-foundations/cells-genomes-dna-chromosomes","docId":"biological-foundations/cells-genomes-dna-chromosomes"},{"type":"link","label":"The Central Dogma","href":"/docs/biological-foundations/the-central-dogma","docId":"biological-foundations/the-central-dogma"},{"type":"link","label":"Transcription","href":"/docs/biological-foundations/transcription","docId":"biological-foundations/transcription"},{"type":"link","label":"Translation","href":"/docs/biological-foundations/translation","docId":"biological-foundations/translation"},{"type":"link","label":"Genotype, Phenotype and Haplotype","href":"/docs/biological-foundations/genotype-phenotype-haplotype","docId":"biological-foundations/genotype-phenotype-haplotype"}],"href":"/docs/biological-foundations/"},{"type":"category","label":"Genomic Variation","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Baselines for Variation","href":"/docs/genomic-variation/baselines-for-variation","docId":"genomic-variation/baselines-for-variation"},{"type":"link","label":"Single Nucleotide Variation","href":"/docs/genomic-variation/single-nucleotide-variation","docId":"genomic-variation/single-nucleotide-variation"},{"type":"link","label":"Insertions and Deletions","href":"/docs/genomic-variation/insertions-and-deletions","docId":"genomic-variation/insertions-and-deletions"},{"type":"link","label":"Structural Variation","href":"/docs/genomic-variation/structural-variation","docId":"genomic-variation/structural-variation"},{"type":"link","label":"Copy Number Variation","href":"/docs/genomic-variation/copy-number-variation","docId":"genomic-variation/copy-number-variation"},{"type":"link","label":"Variant Equivalence","href":"/docs/genomic-variation/variant-equivalence","docId":"genomic-variation/variant-equivalence"},{"type":"link","label":"Summary","href":"/docs/genomic-variation/summary","docId":"genomic-variation/summary"}],"href":"/docs/genomic-variation/"},{"type":"category","label":"Next-generation Sequencing","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Sample Collection","href":"/docs/next-generation-sequencing/sample-collection","docId":"next-generation-sequencing/sample-collection"},{"type":"link","label":"Next-Generation Sequencing","href":"/docs/next-generation-sequencing/next-generation-sequencing","docId":"next-generation-sequencing/next-generation-sequencing"},{"type":"link","label":"Types of Sequencing","href":"/docs/next-generation-sequencing/types-of-sequencing","docId":"next-generation-sequencing/types-of-sequencing"},{"type":"link","label":"Read Mapping","href":"/docs/next-generation-sequencing/read-mapping","docId":"next-generation-sequencing/read-mapping"}],"href":"/docs/next-generation-sequencing/"},{"type":"category","label":"Genomic Files Formats","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Compression and BGZF","href":"/docs/genomic-file-formats/compression-and-BGZF","docId":"genomic-file-formats/compression-and-BGZF"},{"type":"link","label":"FASTQ Files","href":"/docs/genomic-file-formats/FASTQ-files","docId":"genomic-file-formats/FASTQ-files"},{"type":"link","label":"BAM Files","href":"/docs/genomic-file-formats/BAM-files","docId":"genomic-file-formats/BAM-files"},{"type":"link","label":"VCF Files","href":"/docs/genomic-file-formats/VCF-files","docId":"genomic-file-formats/VCF-files"}],"href":"/docs/genomic-file-formats/"},{"type":"category","label":"Engineering Ecosystem","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Data Sources","href":"/docs/engineering-ecosystem/data-sources","docId":"engineering-ecosystem/data-sources"},{"type":"link","label":"Common Genomics Tools","href":"/docs/engineering-ecosystem/common-genomics-tools","docId":"engineering-ecosystem/common-genomics-tools"},{"type":"link","label":"Worked Example","href":"/docs/engineering-ecosystem/end-to-end-example","docId":"engineering-ecosystem/end-to-end-example"}],"href":"/docs/engineering-ecosystem/"},{"type":"category","label":"Advanced Topics","collapsible":true,"collapsed":true,"items":[{"type":"link","label":"Alternative Splicing","href":"/docs/advanced-topics/alternative-splicing","docId":"advanced-topics/alternative-splicing"},{"type":"link","label":"Cancer Development and Control","href":"/docs/advanced-topics/cancer-primer","docId":"advanced-topics/cancer-primer"}],"href":"/docs/advanced-topics/"}]},"docs":{"advanced-topics/alternative-splicing":{"id":"advanced-topics/alternative-splicing","title":"Alternative Splicing","description":"Alternative Splicing","sidebar":"tutorialSidebar"},"advanced-topics/cancer-primer":{"id":"advanced-topics/cancer-primer","title":"Cancer Development and Control","description":"Since any two people share 99.6%--99.9% of their nucleotide sequence, genomic analysts generally focus on the remaining 0.1%--0.4%. While genomic variations may be beneficial or without effect, these variations are implicated in the genesis and progression of many diseases, including many cancers. Historically, cancer researchers focused on \\"driver\\" mutations that negatively affect cell function, causing or influencing cancer. More recent work is broader, including beneficial mutations that may prevent or slow cancer.","sidebar":"tutorialSidebar"},"advanced-topics/index":{"id":"advanced-topics/index","title":"Advanced Topics","description":"You have now completed the main text of the guide\u2014congrats! In this section, we include some guides that didn\'t quite fit into the flow of the main text. Each chapter here is a standalone section and should be considered more experimental in nature. As time goes on and content is developed within these chapters, it\'s possible we will move them into the flow of the main text.","sidebar":"tutorialSidebar"},"biological-foundations/cells-genomes-dna-chromosomes":{"id":"biological-foundations/cells-genomes-dna-chromosomes","title":"Cells, Chromosomes, and Genomes","description":"Introduction","sidebar":"tutorialSidebar"},"biological-foundations/genotype-phenotype-haplotype":{"id":"biological-foundations/genotype-phenotype-haplotype","title":"Genotype, Phenotype and Haplotype","description":"Genotype","sidebar":"tutorialSidebar"},"biological-foundations/index":{"id":"biological-foundations/index","title":"Biological Foundations","description":"We\'ll start with the biological foundations relevant to genomics, which is","sidebar":"tutorialSidebar"},"biological-foundations/the-central-dogma":{"id":"biological-foundations/the-central-dogma","title":"The Central Dogma","description":"Introduction","sidebar":"tutorialSidebar"},"biological-foundations/transcription":{"id":"biological-foundations/transcription","title":"Transcription","description":"Genomic Regions and Transcription","sidebar":"tutorialSidebar"},"biological-foundations/translation":{"id":"biological-foundations/translation","title":"Translation","description":"Decoding the Transcript","sidebar":"tutorialSidebar"},"engineering-ecosystem/common-genomics-tools":{"id":"engineering-ecosystem/common-genomics-tools","title":"Common Genomics Tools","description":"Bioinformatics programmers rely on a growing array of tools written by research scientists. These tools were developed for diverse purposes and were written in different languages. The tools run at different levels of efficiency and differ in reliability. Over time the community of bioinformaticians have built up resources to identify the best packages and handle any quirks or deficiencies they may have. Online resources like GitHub, StackOverflow, and Biostars answer questions and offer community support. Further, developers have provided new tools and methods to ease common problems.","sidebar":"tutorialSidebar"},"engineering-ecosystem/data-sources":{"id":"engineering-ecosystem/data-sources","title":"Data Sources","description":"One of the more challenging aspects of working with genomics data can be finding real","sidebar":"tutorialSidebar"},"engineering-ecosystem/end-to-end-example":{"id":"engineering-ecosystem/end-to-end-example","title":"Worked Example","description":"The following guide is intended to be a simple end-to-end workflow you can run","sidebar":"tutorialSidebar"},"engineering-ecosystem/index":{"id":"engineering-ecosystem/index","title":"Engineering Ecosystem","description":"TODO!","sidebar":"tutorialSidebar"},"genomic-file-formats/BAM-files":{"id":"genomic-file-formats/BAM-files","title":"BAM Files","description":"The\xa0Sequence Alignment/Map\xa0(or SAM) format is the most common output file format after the read alignment. Typically, these files will be encoded in a binary format, bgzipped, indexed, at which point the file is referred to as a\xa0Binary Alignment/Map Format\xa0or BAM file. Each BAM file contains a header (typically used for storing metadata) followed by millions of reads, stored as one read per line. Each read contains all of the information available in the FASTQ file (read name, sequence, associated quality score), as well as additional information added by the aligner, including where the read maps to the reference genome, whether the read was produced by PCR or optical duplication, and the quality of the alignment assigned by the mapping software.","sidebar":"tutorialSidebar"},"genomic-file-formats/compression-and-BGZF":{"id":"genomic-file-formats/compression-and-BGZF","title":"Compression and BGZF","description":"Overview","sidebar":"tutorialSidebar"},"genomic-file-formats/FASTQ-files":{"id":"genomic-file-formats/FASTQ-files","title":"FASTQ Files","description":"For most workflows in computational genomics, the input files are in the FASTQ","sidebar":"tutorialSidebar"},"genomic-file-formats/index":{"id":"genomic-file-formats/index","title":"Genomic File Formats","description":"When working with genomics data for the first time, many engineers are surprised","sidebar":"tutorialSidebar"},"genomic-file-formats/VCF-files":{"id":"genomic-file-formats/VCF-files","title":"VCF Files","description":"The\xa0Variant Call Format\xa0(or VCF) is the most common format for storing variants detected in next-generation sequencing data. Typically, these files will be bgzipped and indexed. Each VCF file contains a header (typically used for storing metadata) and a list of variants. Minimally, each variant contains information about the where the variant lives in the genome and a definition of the change caused by the variant. It also supports storing information about the cohort samples and which samples contain which variants (effectively a\xa0[numvariants] X [numsamples]\xa0matrix with information on every sample, variant pair.)","sidebar":"tutorialSidebar"},"genomic-variation/baselines-for-variation":{"id":"genomic-variation/baselines-for-variation","title":"Baselines for Variation","description":"Introduction","sidebar":"tutorialSidebar"},"genomic-variation/copy-number-variation":{"id":"genomic-variation/copy-number-variation","title":"Copy Number Variation","description":"Copy number variants (CNVs) occur when parts of or entire chromosomes are","sidebar":"tutorialSidebar"},"genomic-variation/index":{"id":"genomic-variation/index","title":"Genomic Variation","description":"During the course of an organism\'s life, changes to its underlying genetic code","sidebar":"tutorialSidebar"},"genomic-variation/insertions-and-deletions":{"id":"genomic-variation/insertions-and-deletions","title":"Insertions and Deletions","description":"Insertions and Deletions are grouped together as a single class of events known","sidebar":"tutorialSidebar"},"genomic-variation/single-nucleotide-variation":{"id":"genomic-variation/single-nucleotide-variation","title":"Single Nucleotide Variation","description":"[Single Nucleotide Variants (SNVs) and Single Nucleotide Polymorphisms","sidebar":"tutorialSidebar"},"genomic-variation/structural-variation":{"id":"genomic-variation/structural-variation","title":"Structural Variation","description":"Living cells are remarkably tough in that they can tolerate large scale changes","sidebar":"tutorialSidebar"},"genomic-variation/summary":{"id":"genomic-variation/summary","title":"Summary","description":"SNVs and Indels form the class of \\"small\\" genetic mutations which is frequently","sidebar":"tutorialSidebar"},"genomic-variation/variant-equivalence":{"id":"genomic-variation/variant-equivalence","title":"Variant Equivalence","description":"A common problem in calling genomic variation is the multitude of ways a","sidebar":"tutorialSidebar"},"next-generation-sequencing/index":{"id":"next-generation-sequencing/index","title":"Next-Generation Sequencing","description":"This chapter discusses the process of how tissue goes from physical material to a file containing the genomic/transcriptomic sequence which can be processed by a computer. This section minimizes the chemical details and focus only on the essentials required to work with data. Understanding how genomic data is acquired helps explain how analyses are done. If you are interested in a more complete (although not exhaustive) description of these processes, see the appendix guide at the end of this section.","sidebar":"tutorialSidebar"},"next-generation-sequencing/next-generation-sequencing":{"id":"next-generation-sequencing/next-generation-sequencing","title":"Next-Generation Sequencing","description":"Once collected, sequencing the sample begins. The sample is removed from","sidebar":"tutorialSidebar"},"next-generation-sequencing/read-mapping":{"id":"next-generation-sequencing/read-mapping","title":"Read Mapping","description":"During the next generation sequencing portion of our workflow, it\'s as if a box","sidebar":"tutorialSidebar"},"next-generation-sequencing/sample-collection":{"id":"next-generation-sequencing/sample-collection","title":"Sample Collection","description":"Before generating sequencing data, first researchers collect and preserve a","sidebar":"tutorialSidebar"},"next-generation-sequencing/types-of-sequencing":{"id":"next-generation-sequencing/types-of-sequencing","title":"Types of Sequencing","description":"Sequencing is laboratory technique that identifies and determines the order of","sidebar":"tutorialSidebar"}}}')}}]);