"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[5718],{3905:function(e,n,t){t.d(n,{Zo:function(){return m},kt:function(){return f}});var r=t(7294);function a(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function i(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);n&&(r=r.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,r)}return t}function o(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?i(Object(t),!0).forEach((function(n){a(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):i(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function s(e,n){if(null==e)return{};var t,r,a=function(e,n){if(null==e)return{};var t,r,a={},i=Object.keys(e);for(r=0;r<i.length;r++)t=i[r],n.indexOf(t)>=0||(a[t]=e[t]);return a}(e,n);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(r=0;r<i.length;r++)t=i[r],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(a[t]=e[t])}return a}var c=r.createContext({}),l=function(e){var n=r.useContext(c),t=n;return e&&(t="function"==typeof e?e(n):o(o({},n),e)),t},m=function(e){var n=l(e.components);return r.createElement(c.Provider,{value:n},e.children)},u={inlineCode:"code",wrapper:function(e){var n=e.children;return r.createElement(r.Fragment,{},n)}},p=r.forwardRef((function(e,n){var t=e.components,a=e.mdxType,i=e.originalType,c=e.parentName,m=s(e,["components","mdxType","originalType","parentName"]),p=l(t),f=a,d=p["".concat(c,".").concat(f)]||p[f]||u[f]||i;return t?r.createElement(d,o(o({ref:n},m),{},{components:t})):r.createElement(d,o({ref:n},m))}));function f(e,n){var t=arguments,a=n&&n.mdxType;if("string"==typeof e||a){var i=t.length,o=new Array(i);o[0]=p;var s={};for(var c in n)hasOwnProperty.call(n,c)&&(s[c]=n[c]);s.originalType=e,s.mdxType="string"==typeof e?e:a,o[1]=s;for(var l=2;l<i;l++)o[l]=t[l];return r.createElement.apply(null,o)}return r.createElement.apply(null,t)}p.displayName="MDXCreateElement"},5381:function(e,n,t){t.r(n),t.d(n,{assets:function(){return m},contentTitle:function(){return c},default:function(){return f},frontMatter:function(){return s},metadata:function(){return l},toc:function(){return u}});var r=t(7462),a=t(3366),i=(t(7294),t(3905)),o=["components"],s={title:"Types of Variation",track:"Genomic Variation"},c=void 0,l={unversionedId:"genomic-variation/types-of-variation",id:"genomic-variation/types-of-variation",title:"Types of Variation",description:"Any difference, large or small, between the sequence of the reference genome and an individual is considered a variant. Each human genome contains at least thousands of variants that are informative enough to identify a person, potential family members and many of their medical risks. Many variants are small, just one base, but some variants alter millions of bases and even whole chromosomes (e.g. an extra copy of chromosome 21 in individuals with Down Syndrome). Variation in the genome takes many forms, including, but not limited to: Single Nucleotide Variants (SNVs), Insertions and Deletions (Indels), Structural Variations (SVs), and Copy Number Variations (CNVs). An important part of computational genomics involves detecting, analyzing and interpreting the relevance of genomic variants to disease.",source:"@site/docs/02-genomic-variation/01-types-of-variation.md",sourceDirName:"02-genomic-variation",slug:"/genomic-variation/types-of-variation",permalink:"/docs/genomic-variation/types-of-variation",editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/02-genomic-variation/01-types-of-variation.md",tags:[],version:"current",sidebarPosition:1,frontMatter:{title:"Types of Variation",track:"Genomic Variation"},sidebar:"tutorialSidebar",previous:{title:"Genomic Variation",permalink:"/docs/genomic-variation/"},next:{title:"Single Nucleotide Variation",permalink:"/docs/genomic-variation/single-nucleotide-variation"}},m={},u=[],p={toc:u};function f(e){var n=e.components,s=(0,a.Z)(e,o);return(0,i.kt)("wrapper",(0,r.Z)({},p,s,{components:n,mdxType:"MDXLayout"}),(0,i.kt)("p",null,"Any difference, large or small, between the sequence of the reference genome and an individual is considered a ",(0,i.kt)("strong",{parentName:"p"},"variant.")," Each human genome contains at least thousands of variants that are informative enough to identify a person, potential family members and many of their medical risks. Many variants are small, just one base, but some variants alter millions of bases and even whole chromosomes (e.g. an extra copy of chromosome 21 in individuals with Down Syndrome). Variation in the genome takes many forms, including, but not limited to: Single Nucleotide Variants (",(0,i.kt)("strong",{parentName:"p"},"SNVs"),"), Insertions and Deletions (",(0,i.kt)("strong",{parentName:"p"},"Indels"),"), Structural Variations (",(0,i.kt)("strong",{parentName:"p"},"SVs"),"), and Copy Number Variations (",(0,i.kt)("strong",{parentName:"p"},"CNVs"),"). An important part of computational genomics involves detecting, analyzing and interpreting the relevance of genomic variants to disease."),(0,i.kt)("p",null,(0,i.kt)("img",{src:t(6995).Z,width:"1000",height:"726"})))}f.isMDXComponent=!0},6995:function(e,n,t){n.Z=t.p+"assets/images/3.1-Small-Variations-d16433e0b75753925f254e545e7de2b5.jpg"}}]);