"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[968],{3905:function(e,t,n){n.d(t,{Zo:function(){return m},kt:function(){return u}});var o=n(7294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function r(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);t&&(o=o.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,o)}return n}function s(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?r(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):r(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function i(e,t){if(null==e)return{};var n,o,a=function(e,t){if(null==e)return{};var n,o,a={},r=Object.keys(e);for(o=0;o<r.length;o++)n=r[o],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);for(o=0;o<r.length;o++)n=r[o],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var l=o.createContext({}),c=function(e){var t=o.useContext(l),n=t;return e&&(n="function"==typeof e?e(t):s(s({},t),e)),n},m=function(e){var t=c(e.components);return o.createElement(l.Provider,{value:t},e.children)},d={inlineCode:"code",wrapper:function(e){var t=e.children;return o.createElement(o.Fragment,{},t)}},p=o.forwardRef((function(e,t){var n=e.components,a=e.mdxType,r=e.originalType,l=e.parentName,m=i(e,["components","mdxType","originalType","parentName"]),p=c(n),u=a,f=p["".concat(l,".").concat(u)]||p[u]||d[u]||r;return n?o.createElement(f,s(s({ref:t},m),{},{components:n})):o.createElement(f,s({ref:t},m))}));function u(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var r=n.length,s=new Array(r);s[0]=p;var i={};for(var l in t)hasOwnProperty.call(t,l)&&(i[l]=t[l]);i.originalType=e,i.mdxType="string"==typeof e?e:a,s[1]=i;for(var c=2;c<r;c++)s[c]=n[c];return o.createElement.apply(null,s)}return o.createElement.apply(null,n)}p.displayName="MDXCreateElement"},8648:function(e,t,n){n.r(t),n.d(t,{assets:function(){return m},contentTitle:function(){return l},default:function(){return u},frontMatter:function(){return i},metadata:function(){return c},toc:function(){return d}});var o=n(7462),a=n(3366),r=(n(7294),n(3905)),s=["components"],i={title:"Common Genomics Tools",track:"Engineering Ecosystem"},l=void 0,c={unversionedId:"engineering-ecosystem/common-genomics-tools",id:"engineering-ecosystem/common-genomics-tools",title:"Common Genomics Tools",description:"Bioinformatics programmers rely on a growing array of tools written by research scientists. These tools were developed for diverse purposes and were written in different languages. The tools run at different levels of efficiency and differ in reliability. Over time the community of bioinformaticians have built up resources to identify the best packages and handle any quirks or deficiencies they may have. Online resources like GitHub, StackOverflow, and Biostars answer questions and offer community support. Further, developers have provided new tools and methods to ease common problems.",source:"@site/docs/05-engineering-ecosystem/02-common-genomics-tools.md",sourceDirName:"05-engineering-ecosystem",slug:"/engineering-ecosystem/common-genomics-tools",permalink:"/docs/engineering-ecosystem/common-genomics-tools",draft:!1,editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/05-engineering-ecosystem/02-common-genomics-tools.md",tags:[],version:"current",sidebarPosition:2,frontMatter:{title:"Common Genomics Tools",track:"Engineering Ecosystem"},sidebar:"tutorialSidebar",previous:{title:"Data Sources",permalink:"/docs/engineering-ecosystem/data-sources"},next:{title:"Worked Example",permalink:"/docs/engineering-ecosystem/end-to-end-example"}},m={},d=[{value:"Common Tools",id:"common-tools",level:2},{value:"BLAST",id:"blast",level:3},{value:"BLAT",id:"blat",level:3},{value:"BWA",id:"bwa",level:3},{value:"STAR",id:"star",level:3},{value:"Samtools",id:"samtools",level:3},{value:"Tabix",id:"tabix",level:3},{value:"Future Tools",id:"future-tools",level:2}],p={toc:d};function u(e){var t=e.components,n=(0,a.Z)(e,s);return(0,r.kt)("wrapper",(0,o.Z)({},p,n,{components:t,mdxType:"MDXLayout"}),(0,r.kt)("p",null,"Bioinformatics programmers rely on a growing array of tools written by research scientists. These tools were developed for diverse purposes and were written in different languages. The tools run at different levels of efficiency and differ in reliability. Over time the community of bioinformaticians have built up resources to identify the best packages and handle any quirks or deficiencies they may have. Online resources like GitHub, StackOverflow, and Biostars answer questions and offer community support. Further, developers have provided new tools and methods to ease common problems."),(0,r.kt)("p",null,'A common problem everyone faces is the installation of unfamiliar packages. While there is no perfect system for coordinating the installation these packages, Bioconda comes close. Bioconda is channel or repository of bioinformatics software "recipes" that can be built into conda packages.\xa0conda\xa0is a package manager that is language and platform independent and is designed to simplify the distribution, installation and management of software packages.\xa0conda\xa0helps end users easily install packages from other repositories, such as pip, CRAN, CPAN, and Bioconductor. We strongly recommend the use of\xa0bioconda, since it is emerging as the modern way to install and work with genomics tools.'),(0,r.kt)("h2",{id:"common-tools"},"Common Tools"),(0,r.kt)("h3",{id:"blast"},"BLAST"),(0,r.kt)("p",null,"One of the first software packages to align sequences was\xa0",(0,r.kt)("a",{parentName:"p",href:"https://blast.ncbi.nlm.nih.gov/Blast.cgi"},"BLAST"),". BLAST stands for\xa0",(0,r.kt)("strong",{parentName:"p"},"B"),"asic\xa0",(0,r.kt)("strong",{parentName:"p"},"L"),"ocal\xa0",(0,r.kt)("strong",{parentName:"p"},"A"),"lignment\xa0",(0,r.kt)("strong",{parentName:"p"},"S"),"earch\xa0",(0,r.kt)("strong",{parentName:"p"},"T"),'ool. It is used to find biological similarity between a query sequence and database of sequences. BLAST comes in different forms that can test DNA, RNA or protein sequences and was developed to improve alignment speed which is necessary to perform the large-scale scans. BLAST uses a rule-based similarity metric that is an approximation of slower more exact methods so BLAST is not guaranteed to find the best alignment and may miss potential alignments. It works by first finding short sequences or "words" from the query sequence, ignoring low complexity words. Then, BLAST uses the best words as "seeds" to find likely sequences. Only sequences in the database that have exact seed matches are retained and tested. This greatly reduces the search space speeding up the test. For matched sequences, the region around the seed is extended in both directions and scored for similarity with the query sequence. If the overall score increases, it is extended further; if not, the extension stops. BLAST retains the best score for each test sequence and reports those that pass a statistical test. BLAST has been used extensively to identify genes with similar functions across multiple species.'),(0,r.kt)("p",null,"For more information, see\xa0",(0,r.kt)("a",{parentName:"p",href:"https://blast.ncbi.nlm.nih.gov/Blast.cgi"},"the NCBI\\'s dedicated page to BLAST"),"."),(0,r.kt)("h3",{id:"blat"},"BLAT"),(0,r.kt)("p",null,"The\xa0",(0,r.kt)("strong",{parentName:"p"},"B"),"LAST-",(0,r.kt)("strong",{parentName:"p"},"l"),"ike\xa0",(0,r.kt)("strong",{parentName:"p"},"A"),"lignment\xa0",(0,r.kt)("strong",{parentName:"p"},"T"),"ool (BLAT) is an alignment tool that works like BLAST but is designed to test against genomes rather than discrete sequences. Like BLAST, BLAT tests for short perfect or near-perfect matches and works well matching sequences with a high degree of similarity. BLAT gets its speed by using an indexed version of the genome or database which it retains in memory. BLAT combines more HSPs into larger alignments compared to BLAST."),(0,r.kt)("h3",{id:"bwa"},"BWA"),(0,r.kt)("p",null,"Though foundational in the field, BLAST and BLAT were not designed to handle the large-scale alignment of millions of short sequences to a reference genome. The\xa0",(0,r.kt)("strong",{parentName:"p"},"B"),"urrows\xa0",(0,r.kt)("strong",{parentName:"p"},"W"),"heeler\xa0",(0,r.kt)("strong",{parentName:"p"},"A"),"ligner, or\xa0bwa, was developed to do exactly this task. This software relies on the\xa0",(0,r.kt)("strong",{parentName:"p"},"B"),"urrows\xa0",(0,r.kt)("strong",{parentName:"p"},"W"),"heeler\xa0",(0,r.kt)("strong",{parentName:"p"},"T"),"ransform (BWT) that changes strings into sorted array of all suffixes of a string. This is a space saving reversible transform that speeds up alignment so that large scale genomic studies can be completed in a timely manner. The reference genome is transformed once and is then used repeatedly for alignment tasks. The software package BWA includes methods for mapping single and paired-end reads and is commonly used for DNA mapping."),(0,r.kt)("p",null,"BWA is widely considered to be the standard for mapping DNA sequences to a reference genome---learn more about it at\xa0",(0,r.kt)("a",{parentName:"p",href:"http://bio-bwa.sourceforge.net/"},"the associated documentation pages"),"."),(0,r.kt)("h3",{id:"star"},"STAR"),(0,r.kt)("p",null,"STAR is an RNA-seq aligner for transcripts that incorporates information about splicing. STAR uses knowledge about the location of exons to index the genome and refine the search for transcript reads. This means that reads that span two exons across a splice site can be accurately mapped. STAR was developed for ENCODE, a public project designed to identify the functional elements in the human genome."),(0,r.kt)("p",null,'STAR "2-pass mapping" is widely considered to be the standard for mapping RNA sequence to a reference genome ---learn more about it at\xa0',(0,r.kt)("a",{parentName:"p",href:"https://github.com/alexdobin/STAR/blob/master/doc/STARmanual.pdf"},"the associated documentation pages"),"."),(0,r.kt)("h3",{id:"samtools"},"Samtools"),(0,r.kt)("p",null,"Samtools is a set of utilities used to perform common operations on SAM/BAM files. Samtools has functions to view, sort, index, merge and assess the quality of aligned sequences contained in a SAM/BAM. Further, samtools can easily transform between SAM/BAM/CRAM files. Samtools is widely used today and is part of every genomic engineer\\'s toolkit: find out more about samtools at\xa0",(0,r.kt)("a",{parentName:"p",href:"http://www.htslib.org/doc/samtools.html"},"the associated documentation pages"),"."),(0,r.kt)("h3",{id:"tabix"},"Tabix"),(0,r.kt)("p",null,"Tabix is a generic tool for indexing tab-delimited text files allowing efficient seek and retrieval. Tabix was written by the author of samtools and can search through bgzipped compressed files. The indexes it produces allow genomic viewers to efficiently view local regions. Tabix indexes sorted tab-delimited data into large intervals called ",(0,r.kt)("strong",{parentName:"p"},"bins"),". Searches for records begin by looking in bins that overlap the query interval."),(0,r.kt)("h2",{id:"future-tools"},"Future Tools"),(0,r.kt)("ul",null,(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Picard")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"bedtools")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"Python libraries"),(0,r.kt)("ul",{parentName:"li"},(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"pysam")),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"BioPython")))),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"R libraries"),(0,r.kt)("ul",{parentName:"li"},(0,r.kt)("li",{parentName:"ul"},"BioConductor"))),(0,r.kt)("li",{parentName:"ul"},(0,r.kt)("p",{parentName:"li"},"PERL libraries"),(0,r.kt)("ul",{parentName:"li"},(0,r.kt)("li",{parentName:"ul"},"BioPerl")))))}u.isMDXComponent=!0}}]);