"use strict";(self.webpackChunklearngenomics_dev=self.webpackChunklearngenomics_dev||[]).push([[75],{3905:function(e,n,t){t.d(n,{Zo:function(){return u},kt:function(){return p}});var a=t(7294);function r(e,n,t){return n in e?Object.defineProperty(e,n,{value:t,enumerable:!0,configurable:!0,writable:!0}):e[n]=t,e}function i(e,n){var t=Object.keys(e);if(Object.getOwnPropertySymbols){var a=Object.getOwnPropertySymbols(e);n&&(a=a.filter((function(n){return Object.getOwnPropertyDescriptor(e,n).enumerable}))),t.push.apply(t,a)}return t}function o(e){for(var n=1;n<arguments.length;n++){var t=null!=arguments[n]?arguments[n]:{};n%2?i(Object(t),!0).forEach((function(n){r(e,n,t[n])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(t)):i(Object(t)).forEach((function(n){Object.defineProperty(e,n,Object.getOwnPropertyDescriptor(t,n))}))}return e}function s(e,n){if(null==e)return{};var t,a,r=function(e,n){if(null==e)return{};var t,a,r={},i=Object.keys(e);for(a=0;a<i.length;a++)t=i[a],n.indexOf(t)>=0||(r[t]=e[t]);return r}(e,n);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);for(a=0;a<i.length;a++)t=i[a],n.indexOf(t)>=0||Object.prototype.propertyIsEnumerable.call(e,t)&&(r[t]=e[t])}return r}var c=a.createContext({}),l=function(e){var n=a.useContext(c),t=n;return e&&(t="function"==typeof e?e(n):o(o({},n),e)),t},u=function(e){var n=l(e.components);return a.createElement(c.Provider,{value:n},e.children)},g={inlineCode:"code",wrapper:function(e){var n=e.children;return a.createElement(a.Fragment,{},n)}},d=a.forwardRef((function(e,n){var t=e.components,r=e.mdxType,i=e.originalType,c=e.parentName,u=s(e,["components","mdxType","originalType","parentName"]),d=l(t),p=r,h=d["".concat(c,".").concat(p)]||d[p]||g[p]||i;return t?a.createElement(h,o(o({ref:n},u),{},{components:t})):a.createElement(h,o({ref:n},u))}));function p(e,n){var t=arguments,r=n&&n.mdxType;if("string"==typeof e||r){var i=t.length,o=new Array(i);o[0]=d;var s={};for(var c in n)hasOwnProperty.call(n,c)&&(s[c]=n[c]);s.originalType=e,s.mdxType="string"==typeof e?e:r,o[1]=s;for(var l=2;l<i;l++)o[l]=t[l];return a.createElement.apply(null,o)}return a.createElement.apply(null,t)}d.displayName="MDXCreateElement"},2913:function(e,n,t){t.r(n),t.d(n,{assets:function(){return u},contentTitle:function(){return c},default:function(){return p},frontMatter:function(){return s},metadata:function(){return l},toc:function(){return g}});var a=t(7462),r=t(3366),i=(t(7294),t(3905)),o=["components"],s={title:"Types of Sequencing",track:"Next-Generation Sequencing"},c=void 0,l={unversionedId:"next-generation-sequencing/types-of-sequencing",id:"next-generation-sequencing/types-of-sequencing",title:"Types of Sequencing",description:"Sequencing is laboratory technique that identifies and determines the order of",source:"@site/docs/03-next-generation-sequencing/03-types-of-sequencing.md",sourceDirName:"03-next-generation-sequencing",slug:"/next-generation-sequencing/types-of-sequencing",permalink:"/learngenomics.dev/docs/next-generation-sequencing/types-of-sequencing",draft:!1,editUrl:"https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/03-next-generation-sequencing/03-types-of-sequencing.md",tags:[],version:"current",sidebarPosition:3,frontMatter:{title:"Types of Sequencing",track:"Next-Generation Sequencing"},sidebar:"tutorialSidebar",previous:{title:"Next-Generation Sequencing",permalink:"/learngenomics.dev/docs/next-generation-sequencing/next-generation-sequencing"},next:{title:"Read Mapping",permalink:"/learngenomics.dev/docs/next-generation-sequencing/read-mapping"}},u={},g=[{value:"Common Sequence Applications",id:"common-sequence-applications",level:2},{value:"Whole Genome Sequencing",id:"whole-genome-sequencing",level:2},{value:"Whole Exome Sequencing",id:"whole-exome-sequencing",level:2},{value:"Whole Transcriptome Sequencing / RNA-seq",id:"whole-transcriptome-sequencing--rna-seq",level:2},{value:"Targeted Sequencing",id:"targeted-sequencing",level:2}],d={toc:g};function p(e){var n=e.components,s=(0,r.Z)(e,o);return(0,i.kt)("wrapper",(0,a.Z)({},d,s,{components:n,mdxType:"MDXLayout"}),(0,i.kt)("p",null,"Sequencing is laboratory technique that identifies and determines the order of\nbases in a DNA molecule. Each sequencing type is best analyzed by a different\ncomputational method and is interpreted in different way. The scientific goals\nof the research, cost and practical limits determine which types of sequencing\nare chosen. For example, clinical sequencing seeks the time-sensitive detection\nof known mutations, while population studies focus on the discovery of new\nvariants. Often, the type of sequencing ordered is a trade-off between cost and\nthe breadth of information. This section discusses what scientists weigh when\ndeciding which pieces of the genome/exome/transcriptome needs sequencing."),(0,i.kt)("h2",{id:"common-sequence-applications"},"Common Sequence Applications"),(0,i.kt)("table",null,(0,i.kt)("thead",{parentName:"table"},(0,i.kt)("tr",{parentName:"thead"},(0,i.kt)("th",{parentName:"tr",align:null},(0,i.kt)("strong",{parentName:"th"},"Area of the genome")),(0,i.kt)("th",{parentName:"tr",align:null},(0,i.kt)("strong",{parentName:"th"},"Abbreviations")),(0,i.kt)("th",{parentName:"tr",align:null},(0,i.kt)("strong",{parentName:"th"},"Description")))),(0,i.kt)("tbody",{parentName:"table"},(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Whole Genome Sequencing"),(0,i.kt)("td",{parentName:"tr",align:null},"WGS"),(0,i.kt)("td",{parentName:"tr",align:null},"Characterizes the entire genome reading sequences from DNA")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Whole Exome Sequencing"),(0,i.kt)("td",{parentName:"tr",align:null},"WES, WXS"),(0,i.kt)("td",{parentName:"tr",align:null},"Characterizes the exonic, or coding, regions of the genome reading sequences from DNA")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Transcriptome Sequencing"),(0,i.kt)("td",{parentName:"tr",align:null},"WTS, RNA-seq"),(0,i.kt)("td",{parentName:"tr",align:null},"Characterizes the exonic regions that are actively expressed in the cell reading sequences from RNA")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Targeted Sequencing"),(0,i.kt)("td",{parentName:"tr",align:null},"Validation Capture, Valcap"),(0,i.kt)("td",{parentName:"tr",align:null},"Characterizes a chosen region from DNA, can be a coding or non-coding region. The region can be coding or non-coding and is typically focused, high-depth sequencing for validating variants")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"ChIP seq"),(0,i.kt)("td",{parentName:"tr",align:null},"ChIP-seq"),(0,i.kt)("td",{parentName:"tr",align:null},"Characterizes any sequence the genomic locations where a protein interacts with DNA, both coding and non-coding")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"ATAC sequencing"),(0,i.kt)("td",{parentName:"tr",align:null},"ATAC-seq"),(0,i.kt)("td",{parentName:"tr",align:null},'Characterizes any sequence where DNA does not interact with proteins or is "open", both coding and non-coding')),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Hi-C"),(0,i.kt)("td",{parentName:"tr",align:null},"Hi-C"),(0,i.kt)("td",{parentName:"tr",align:null},"Characterizes the genomic locations where chromosomes come in close contact with each other. Typically used in characterizing 3D interactions of DNA")),(0,i.kt)("tr",{parentName:"tbody"},(0,i.kt)("td",{parentName:"tr",align:null},"Single Cell Sequencing"),(0,i.kt)("td",{parentName:"tr",align:null},"scDNA-seq, scRNA-seq"),(0,i.kt)("td",{parentName:"tr",align:null},"DNA or RNA sequencing of individual cells, captures granularity at an individual cell level rather than a mixture of cells, can be coding or non-coding with lower coverage")))),(0,i.kt)("h2",{id:"whole-genome-sequencing"},"Whole Genome Sequencing"),(0,i.kt)("p",null,"If cost was not an issue, whole genome sequencing (",(0,i.kt)("strong",{parentName:"p"},"WGS"),') would be the\nstandard approach for detecting variants. The advantage of WGS is a complete\nsurvey of the DNA, allowing the researcher to detect all the genomic variation\nfor a given sample. WGS is needed when exploring how variations in non-coding\nregions influence the cells. Further, WGS is needed to detect the scope of\nvariants that affect large regions of the chromosome (Structural Variation and\nCopy Number Variation, which are discussed in the "Genomic Variation" chapter).\nThe disadvantage, other than cost, is that many of variations are not relevant\nto disease or the research question and need to be carefully filtered out. WGS\nis used in basic exploratory research and increasingly elsewhere as costs\ndrop.'),(0,i.kt)("p",null,(0,i.kt)("img",{alt:"Figure depicting the entire genome being sequenced using NGS.",src:t(5204).Z,width:"3071",height:"935"})),(0,i.kt)("h2",{id:"whole-exome-sequencing"},"Whole Exome Sequencing"),(0,i.kt)("p",null,"Whole exome sequencing (",(0,i.kt)("strong",{parentName:"p"},"WES"),") captures the coding sequences of known genes\nfrom DNA and is a focused, cost-effective approach to characterizing what\nprotein byproducts might be made. WES is commonly used to rapidly detect\nmutations in tumor samples so that treatments can be tailored to the individual.\nIt is less commonly used in basic research due to its limited scope. The\nexceptions are large scale population studies with thousands of samples where\nwhole genome approaches are too expensive."),(0,i.kt)("p",null,(0,i.kt)("img",{alt:"Figure depicting the exome being sequenced using NGS.",src:t(5259).Z,width:"3071",height:"1000"})),(0,i.kt)("h2",{id:"whole-transcriptome-sequencing--rna-seq"},"Whole Transcriptome Sequencing / RNA-seq"),(0,i.kt)("p",null,"Whole transcriptome sequencing, or\xa0",(0,i.kt)("strong",{parentName:"p"},"RNA-seq"),", captures the working recipes of\ngenes as they float around a cell. This sequencing type gives the researcher a\nwindow into the mixture of proteins being created by the cell when the sample\nwas collected. Mutations and natural variations, especially those which alter\nprotein byproducts, can be detected using RNA-seq. Additionally, a common goal\nof RNA-seq is to quantify the transcripts of active genes to understand which\ngenes are turned on and off in response to experimental treatments, disease or\ndevelopmental changes in the cell."),(0,i.kt)("p",null,(0,i.kt)("img",{alt:"Figure depicting the messenger RNA sequenced using NGS.",src:t(1665).Z,width:"3071",height:"999"})),(0,i.kt)("h2",{id:"targeted-sequencing"},"Targeted Sequencing"),(0,i.kt)("p",null,(0,i.kt)("strong",{parentName:"p"},"Targeted sequencing")," or validation capture sequencing (",(0,i.kt)("strong",{parentName:"p"},"Valcap"),") captures\npreselected regions in the genome based on experimental design. Often targeted\nsequencing is used to capture a subset of important genes (called ",(0,i.kt)("strong",{parentName:"p"},"gene\npanels"),") in standardized tests when cost and time are limited Alternatively,\ntargeted sequencing is frequently used to gain greater insight into a region\nobserved through a whole genome or whole exome study where there is less read\ndepth. The method can be tailored for coding or non-coding variants."),(0,i.kt)("p",null,(0,i.kt)("img",{alt:"Figure depicting targeted regions being deep sequenced using NGS.",src:t(8955).Z,width:"2447",height:"1382"})))}p.isMDXComponent=!0},5204:function(e,n,t){n.Z=t.p+"assets/images/3.3-Whole-Genome-Sequencing-6a451d5ee2e6d2fa423b40e26ec468e6.jpg"},5259:function(e,n,t){n.Z=t.p+"assets/images/3.4-Whole-Exome-Sequencing-6337dd6cdab3a46da2fce75f4c759cb2.jpg"},1665:function(e,n,t){n.Z=t.p+"assets/images/3.5-RNA-Sequencing-d837cb25d70cbef03d611ccbabce07ff.jpg"},8955:function(e,n,t){n.Z=t.p+"assets/images/3.6-Targeted-Sequencing-dd4f748b80eb4f7abb3e342f690b2144.jpg"}}]);