<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-next-generation-sequencing/next-generation-sequencing">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.18">
<link rel="preconnect" href="https://www.google-analytics.com">
<link rel="preconnect" href="https://www.googletagmanager.com">
<script async src="https://www.googletagmanager.com/gtag/js?id=G-6XD8R1VWBJ"></script>
<script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-6XD8R1VWBJ",{anonymize_ip:!0})</script><title data-rh="true">Next-Generation Sequencing | Introduction to Genomics for Engineers</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://learngenomics.dev/docs/next-generation-sequencing/next-generation-sequencing"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Next-Generation Sequencing | Introduction to Genomics for Engineers"><meta data-rh="true" name="description" content="Once collected, sequencing the sample begins. The sample is removed from storage, and a portion is taken to prepare what is called a &quot;library&quot; for sequencing. This is always performed in a wet lab environment, and the chemical methods can be quite complex."><meta data-rh="true" property="og:description" content="Once collected, sequencing the sample begins. The sample is removed from storage, and a portion is taken to prepare what is called a &quot;library&quot; for sequencing. This is always performed in a wet lab environment, and the chemical methods can be quite complex."><link data-rh="true" rel="icon" href="/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://learngenomics.dev/docs/next-generation-sequencing/next-generation-sequencing"><link data-rh="true" rel="alternate" href="https://learngenomics.dev/docs/next-generation-sequencing/next-generation-sequencing" hreflang="en"><link data-rh="true" rel="alternate" href="https://learngenomics.dev/docs/next-generation-sequencing/next-generation-sequencing" hreflang="x-default"><link rel="stylesheet" href="/assets/css/styles.7d27f8ab.css">
<link rel="preload" href="/assets/js/runtime~main.d0bd14fb.js" as="script">
<link rel="preload" href="/assets/js/main.9589c47f.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/"><div class="navbar__logo"><img src="/img/learngenomics-logo.svg" alt="learngenomics.dev logo: a DNA helix beside five lines of code." class="themedImage_W2Cr themedImage--light_TfLj"><img src="/img/learngenomics-logo.svg" alt="learngenomics.dev logo: a DNA helix beside five lines of code." class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title">learngenomics.dev</b></a><a class="navbar__item navbar__link navbar__link--active" href="/docs/foreword">Guide</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/learngenomics/learngenomics.dev" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper"><div class="docPage_P2Lg"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><aside class="theme-doc-sidebar-container docSidebarContainer_rKC_"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/docs/foreword">Foreword</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/biological-foundations/">Biological Foundations</a><button aria-label="Toggle the collapsible sidebar category &#x27;Biological Foundations&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/genomic-variation/">Genomic Variation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genomic Variation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/docs/next-generation-sequencing/">Next-generation Sequencing</a><button aria-label="Toggle the collapsible sidebar category &#x27;Next-generation Sequencing&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/docs/next-generation-sequencing/sample-collection">Sample Collection</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/docs/next-generation-sequencing/next-generation-sequencing">Next-Generation Sequencing</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/docs/next-generation-sequencing/read-mapping">Read Mapping</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/genomic-file-formats/">Genomic Files Formats</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genomic Files Formats&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/engineering-ecosystem/">Engineering Ecosystem</a><button aria-label="Toggle the collapsible sidebar category &#x27;Engineering Ecosystem&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/advanced-topics/">Advanced Topics</a><button aria-label="Toggle the collapsible sidebar category &#x27;Advanced Topics&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_TCnq"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a class="breadcrumbs__link" href="/">🏠</a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/docs/next-generation-sequencing/"><span itemprop="name">Next-generation Sequencing</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="item name">Next-Generation Sequencing</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_jdIR theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_Fzxq">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>Next-Generation Sequencing</h1></header><p>Once collected, sequencing the sample begins. The sample is removed from storage, and a portion is taken to prepare what is called a &quot;library&quot; for sequencing. This is always performed in a wet lab environment, and the chemical methods can be quite complex.</p><p>While it sounds easy to read the entire genome by laying all of the chromosomes end-to-end, in practice, there are some problems with this approach. Chiefly, researchers currently lack the biochemical processing abilities to extract to read such long stretches of DNA quickly and accurately. Additionally, next generation sequencing technologies have their own set of limitations, including:</p><ul><li><p>Each nucleotide the sequencer calls is accompanied by a confidence score estimating the how likely the recorded call is true. In short, the sequencer\&#x27;s calls are not perfect.</p></li><li><p>As the sequencer observes each nucleotide along the read, the confidence it has in each new call drops so that if allowed to continue eventually the data quality of the last bases would be too low to use reliably.</p></li><li><p>Sequencers can only read DNA fragments because RNA is too unstable. So biochemical methods are employed to reverse the transcription process and produce a DNA fragment that captures the information in the RNA sequence.</p></li></ul><p>Given these physical limitations, the process, called <strong>shotgun sequencing</strong>, looks more like the following:</p><ol><li><p>The sample&#x27;s DNA is fragmented into billions of smaller DNA pieces that are somewhere between tens to thousands of bases long.</p></li><li><p>This soup of DNA fragments with randomly distributed lengths is then put through a filtering step called &quot;size selection&quot;. Only DNA fragments of a certain length will remain (typically somewhere between 200-500 bp, where bp stands for base pair).</p></li><li><p>Among many other important steps to prepare the DNA library for sequencing is &quot;adapter ligation&quot;. This process involves tacking on special sequences to both ends of the DNA fragment which is required by the sequencing process.</p></li></ol><p><img loading="lazy" src="/assets/images/4.1-Read-on-Flow-Cell-Zoomed-ed0ea55a052a14fd30a44bbd97cae2d3.jpg" width="1000" height="352" class="img_E7b_"></p><ol start="4"><li><p>Optionally, a chemical amplification step can be conducted to increase the amount of genomic material available for sequencing.</p></li><li><p>The sample is loaded onto a flow cell and put into an Illumina sequencer, which is capable of reading millions of DNA fragments and capturing the results digitally.</p></li><li><p>Once the sequencing is complete, the sequencer closes the sequencing session by creating a directory of BCL files (Illumina\&#x27;s proprietary storage format). These are immediately converted to a FASTQ file (standard DNA storage format), typically the &quot;input&quot; to the computational team.</p></li></ol><p><img loading="lazy" src="/assets/images/4.2-Reads-on-Flow-Cell-c72528a8a3f89d7afe95115bac0b5ae6.jpg" width="1000" height="745" class="img_E7b_"></p><p>For our paired-end sequencing example, it is important to note how the DNA sequencer reads the data in step 4. In each 200-500bp DNA fragment, the sequencer begins reading from one of the fragments and drops off somewhere in the middle. It stores all of the nucleotides read from this end of the fragment with their associated confidence scores as &quot;Read 1&quot;. After this is complete, the sequencer reads from the opposite end of the fragment and stores the nucleotide calls and confidence scores known as &quot;Read 2&quot;. Together, the combination of a Read 1 from a DNA fragment and a Read 2 from the other end of the DNA fragment is called a &quot;read pair&quot; as denoted in Figure <!-- -->#<!-- -->.</p><p>After the steps above have completed, the result is a single FASTQ file (single-end sequencing) or a pair of FASTQ files (paired-end). For more information on the FASTQ file format, see its section in the &quot;Genomic File Formats&quot; chapter.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="types-of-sequencing">Types of Sequencing<a class="hash-link" href="#types-of-sequencing" title="Direct link to heading">​</a></h2><p>Sequencing is laboratory technique that identifies and determines the order of bases in a DNA molecule. Each sequencing type is best analyzed by a different computational method and is interpreted in different way. The scientific goals of the research, cost and practical limits determine which types of sequencing are chosen. For example, clinical sequencing seeks the time-sensitive detection of known mutations, while population studies focus on the discovery of new variants. Often, the type of sequencing ordered is a trade-off between cost and the breadth of information. This section discusses what scientists weigh when deciding which pieces of the genome/exome/transcriptome needs sequencing; the &quot;Data Acquisition&quot; section of this guide focuses on how the genome/exome/transcriptome are sequenced.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="common-sequence-applications">Common Sequence Applications<a class="hash-link" href="#common-sequence-applications" title="Direct link to heading">​</a></h2><table><thead><tr><th><strong>Area of the genome</strong></th><th><strong>Abbreviations</strong></th><th><strong>Description</strong></th></tr></thead><tbody><tr><td>Whole Genome Sequencing</td><td>WGS</td><td>Characterizes the entire genome reading sequences from DNA</td></tr><tr><td>Whole Exome Sequencing</td><td>WES, WXS</td><td>Characterizes the exonic, or coding, regions of the genome reading sequences from DNA</td></tr><tr><td>Transcriptome Sequencing</td><td>WTS, RNA-seq</td><td>Characterizes the exonic regions that are actively expressed in the cell reading sequences from RNA</td></tr><tr><td>Targeted Sequencing</td><td>Validation Capture, Valcap</td><td>Characterizes a chosen region from DNA, can be a coding or non-coding region. The region can be coding or non-coding and is typically focused, high-depth sequencing for validating variants</td></tr><tr><td>ChIP seq</td><td>ChIP-seq</td><td>Characterizes any sequence the genomic locations where a protein interacts with DNA, both coding and non-coding</td></tr><tr><td>ATAC sequencing</td><td>ATAC-seq</td><td>Characterizes any sequence where DNA does not interact with proteins or is &quot;open&quot;, both coding and non-coding</td></tr><tr><td>Hi-C</td><td>Hi-C</td><td>Characterizes the genomic locations where chromosomes come in close contact with each other. Typically used in characterizing 3D interactions of DNA</td></tr><tr><td>Single Cell Sequencing</td><td>scDNA-seq, scRNA-seq</td><td>DNA or RNA sequencing of individual cells, captures granularity at an individual cell level rather than a mixture of cells, can be coding or non-coding with lower coverage</td></tr></tbody></table><h2 class="anchor anchorWithStickyNavbar_mojV" id="whole-genome-sequencing">Whole Genome Sequencing<a class="hash-link" href="#whole-genome-sequencing" title="Direct link to heading">​</a></h2><p>If cost was not an issue, whole genome sequencing (<strong>WGS</strong>) would be the standard approach for detecting variants. The advantage of WGS is a complete survey of the DNA, allowing the researcher to detect all the genomic variation for a given sample. WGS is needed when exploring how variations in non-coding regions influence the cells. Further, WGS is needed to detect the scope of variants that affect large regions of the chromosome (Structural Variation and Copy Number Variation, which are discussed in Chapter 3). The disadvantage, other than cost, is that many of variations are not relevant to disease or the research question and need to be carefully filtered out. WGS is used in basic exploratory research and increasingly in the clinic as costs drop.</p><p><img loading="lazy" src="/assets/images/2.1-WGS-d88352e5c737eca2fb4974e54420a7c2.jpg" width="2000" height="607" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="whole-exome-sequencing">Whole Exome Sequencing<a class="hash-link" href="#whole-exome-sequencing" title="Direct link to heading">​</a></h2><p>Whole exome sequencing (<strong>WES</strong>) captures the coding sequences of known genes from DNA and is a focused, cost-effective approach to characterizing what protein byproducts might be made. WES is commonly used in the clinic to rapidly detect mutations in tumor samples so that treatments can be tailored to the individual. It is less commonly used in basic research due to its limited scope. The exceptions are large scale population studies with thousands of samples where whole genome approaches are too expensive.</p><p><img loading="lazy" src="/assets/images/2.2-WES-d54690a3e4a2bcdf075c8481e4ca6f5d.jpg" width="1000" height="326" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="whole-transcriptome-sequencing--rna-seq">Whole Transcriptome Sequencing / RNA-seq<a class="hash-link" href="#whole-transcriptome-sequencing--rna-seq" title="Direct link to heading">​</a></h2><p>Whole transcriptome sequencing, or <strong>RNA-seq</strong>, captures the working recipes of genes as they float around a cell. This sequencing type gives the researcher a window into the mixture of proteins being created by the cell when the sample was collected. Mutations and natural variations, especially those which alter protein byproducts, can be detected using RNA-seq. Additionally, a common goal of RNA-seq is to quantify the transcripts of active genes to understand which genes are turned on and off in response to experimental treatments, disease or developmental changes in the cell.</p><p><img loading="lazy" src="/assets/images/2.3-RNASeq-2a757ea6fd59281d9e647bf88de14d33.jpg" width="2000" height="651" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="targeted-sequencing">Targeted Sequencing<a class="hash-link" href="#targeted-sequencing" title="Direct link to heading">​</a></h2><p><strong>Targeted sequencing</strong> or validation capture sequencing (<strong>Valcap</strong>) captures preselected regions in the genome based on experimental design. Often targeted sequencing is used to capture a subset of important genes (called <strong>gene panels</strong>) in standardized tests when cost and time are limited Alternatively, targeted sequencing is frequently used to gain greater insight into a region observed through a whole genome or whole exome study where there is less read depth. The method can be tailored for coding or non-coding variants.</p><p><img loading="lazy" src="/assets/images/2.4-Targeted-d15a6add663d900051894629579077e9.jpg" width="1000" height="564" class="img_E7b_"></p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/03-next-generation-sequencing/02-next-generation-sequencing.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/docs/next-generation-sequencing/sample-collection"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Sample Collection</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/docs/next-generation-sequencing/read-mapping"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Read Mapping</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#types-of-sequencing" class="table-of-contents__link toc-highlight">Types of Sequencing</a></li><li><a href="#common-sequence-applications" class="table-of-contents__link toc-highlight">Common Sequence Applications</a></li><li><a href="#whole-genome-sequencing" class="table-of-contents__link toc-highlight">Whole Genome Sequencing</a></li><li><a href="#whole-exome-sequencing" class="table-of-contents__link toc-highlight">Whole Exome Sequencing</a></li><li><a href="#whole-transcriptome-sequencing--rna-seq" class="table-of-contents__link toc-highlight">Whole Transcriptome Sequencing / RNA-seq</a></li><li><a href="#targeted-sequencing" class="table-of-contents__link toc-highlight">Targeted Sequencing</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Guide</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/biological-foundations">Biological Foundations</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/genomic-variation">Genomic Variation</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/next-generation-sequencing">Next-generation Sequencing</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/engineering-ecosystem">Engineering Ecosystem</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/advanced-topics">Advanced Topics</a></li></ul></div><div class="col footer__col"><div class="footer__title">Socials</div><ul class="footer__items"><li class="footer__item"><a href="https://github.com/learngenomics/learngenomics.dev" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2022 learngenomics.dev and associated contributors. Built with Docusaurus.</div></div></div></footer></div>
<script src="/assets/js/runtime~main.d0bd14fb.js"></script>
<script src="/assets/js/main.9589c47f.js"></script>
</body>
</html>