<!doctype html>
<html lang="en" dir="ltr" class="docs-wrapper docs-doc-page docs-version-current plugin-docs plugin-id-default docs-doc-id-engineering-ecosystem/genomic-file-formats">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.18">
<link rel="alternate" type="application/rss+xml" href="/blog/rss.xml" title="Introduction to Genomics for Engineers RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/blog/atom.xml" title="Introduction to Genomics for Engineers Atom Feed"><title data-rh="true">Genomic File Formats | Introduction to Genomics for Engineers</title><meta data-rh="true" name="twitter:card" content="summary_large_image"><meta data-rh="true" property="og:url" content="https://learngenomics.dev/docs/engineering-ecosystem/genomic-file-formats"><meta data-rh="true" name="docusaurus_locale" content="en"><meta data-rh="true" name="docsearch:language" content="en"><meta data-rh="true" name="docusaurus_version" content="current"><meta data-rh="true" name="docusaurus_tag" content="docs-default-current"><meta data-rh="true" name="docsearch:version" content="current"><meta data-rh="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-rh="true" property="og:title" content="Genomic File Formats | Introduction to Genomics for Engineers"><meta data-rh="true" name="description" content="When working with genomics data for the first time, many engineers are surprised to find that genomics file types are quite simple in structure. In fact, a majority of universal file formats such as SAM/BAM, VCF, BED, GFF/GTF and many other custom formats are simply tab-delimited text files with some form of compression. Mastery of Unix command line tools such as grep, less, cut, join, sort, sed, and awk are essential to working effectively in this domain."><meta data-rh="true" property="og:description" content="When working with genomics data for the first time, many engineers are surprised to find that genomics file types are quite simple in structure. In fact, a majority of universal file formats such as SAM/BAM, VCF, BED, GFF/GTF and many other custom formats are simply tab-delimited text files with some form of compression. Mastery of Unix command line tools such as grep, less, cut, join, sort, sed, and awk are essential to working effectively in this domain."><link data-rh="true" rel="icon" href="/img/favicon.ico"><link data-rh="true" rel="canonical" href="https://learngenomics.dev/docs/engineering-ecosystem/genomic-file-formats"><link data-rh="true" rel="alternate" href="https://learngenomics.dev/docs/engineering-ecosystem/genomic-file-formats" hreflang="en"><link data-rh="true" rel="alternate" href="https://learngenomics.dev/docs/engineering-ecosystem/genomic-file-formats" hreflang="x-default"><link rel="stylesheet" href="/assets/css/styles.beb074ef.css">
<link rel="preload" href="/assets/js/runtime~main.fe122cfd.js" as="script">
<link rel="preload" href="/assets/js/main.585ef083.js" as="script">
</head>
<body class="navigation-with-keyboard">
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div role="region"><a href="#" class="skipToContent_ZgBM">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/"><div class="navbar__logo"><img src="/img/learngenomics-logo.svg" alt="learngenomics.dev logo: a DNA helix beside five lines of code." class="themedImage_W2Cr themedImage--light_TfLj"><img src="/img/learngenomics-logo.svg" alt="learngenomics.dev logo: a DNA helix beside five lines of code." class="themedImage_W2Cr themedImage--dark_oUvU"></div><b class="navbar__title">learngenomics.dev</b></a><a class="navbar__item navbar__link navbar__link--active" href="/docs/foreword">Guide</a></div><div class="navbar__items navbar__items--right"><a href="https://github.com/learngenomics/learngenomics.dev" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a><div class="toggle_S7eR colorModeToggle_vKtC"><button class="clean-btn toggleButton_rCf9 toggleButtonDisabled_Pu9x" type="button" disabled="" title="Switch between dark and light mode (currently light mode)" aria-label="Switch between dark and light mode (currently light mode)"><svg viewBox="0 0 24 24" width="24" height="24" class="lightToggleIcon_v35p"><path fill="currentColor" d="M12,9c1.65,0,3,1.35,3,3s-1.35,3-3,3s-3-1.35-3-3S10.35,9,12,9 M12,7c-2.76,0-5,2.24-5,5s2.24,5,5,5s5-2.24,5-5 S14.76,7,12,7L12,7z M2,13l2,0c0.55,0,1-0.45,1-1s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S1.45,13,2,13z M20,13l2,0c0.55,0,1-0.45,1-1 s-0.45-1-1-1l-2,0c-0.55,0-1,0.45-1,1S19.45,13,20,13z M11,2v2c0,0.55,0.45,1,1,1s1-0.45,1-1V2c0-0.55-0.45-1-1-1S11,1.45,11,2z M11,20v2c0,0.55,0.45,1,1,1s1-0.45,1-1v-2c0-0.55-0.45-1-1-1C11.45,19,11,19.45,11,20z M5.99,4.58c-0.39-0.39-1.03-0.39-1.41,0 c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0s0.39-1.03,0-1.41L5.99,4.58z M18.36,16.95 c-0.39-0.39-1.03-0.39-1.41,0c-0.39,0.39-0.39,1.03,0,1.41l1.06,1.06c0.39,0.39,1.03,0.39,1.41,0c0.39-0.39,0.39-1.03,0-1.41 L18.36,16.95z M19.42,5.99c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06c-0.39,0.39-0.39,1.03,0,1.41 s1.03,0.39,1.41,0L19.42,5.99z M7.05,18.36c0.39-0.39,0.39-1.03,0-1.41c-0.39-0.39-1.03-0.39-1.41,0l-1.06,1.06 c-0.39,0.39-0.39,1.03,0,1.41s1.03,0.39,1.41,0L7.05,18.36z"></path></svg><svg viewBox="0 0 24 24" width="24" height="24" class="darkToggleIcon_nQuB"><path fill="currentColor" d="M9.37,5.51C9.19,6.15,9.1,6.82,9.1,7.5c0,4.08,3.32,7.4,7.4,7.4c0.68,0,1.35-0.09,1.99-0.27C17.45,17.19,14.93,19,12,19 c-3.86,0-7-3.14-7-7C5,9.07,6.81,6.55,9.37,5.51z M12,3c-4.97,0-9,4.03-9,9s4.03,9,9,9s9-4.03,9-9c0-0.46-0.04-0.92-0.1-1.36 c-0.98,1.37-2.58,2.26-4.4,2.26c-2.98,0-5.4-2.42-5.4-5.4c0-1.81,0.89-3.42,2.26-4.4C12.92,3.04,12.46,3,12,3L12,3z"></path></svg></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper"><div class="docPage_P2Lg"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_RiI4" type="button"></button><aside class="theme-doc-sidebar-container docSidebarContainer_rKC_"><div class="sidebar_RiAD"><nav class="menu thin-scrollbar menu_izAj"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-1 menu__list-item"><a class="menu__link" href="/docs/foreword">Foreword</a></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/biological-foundations/">Biological Foundations</a><button aria-label="Toggle the collapsible sidebar category &#x27;Biological Foundations&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/genomic-variation/">Genomic Variation</a><button aria-label="Toggle the collapsible sidebar category &#x27;Genomic Variation&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/next-generation-sequencing/">Next-generation Sequencing</a><button aria-label="Toggle the collapsible sidebar category &#x27;Next-generation Sequencing&#x27;" type="button" class="clean-btn menu__caret"></button></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active" aria-expanded="true" href="/docs/engineering-ecosystem/">Engineering Ecosystem</a><button aria-label="Toggle the collapsible sidebar category &#x27;Engineering Ecosystem&#x27;" type="button" class="clean-btn menu__caret"></button></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/docs/engineering-ecosystem/genomic-file-formats">Genomic File Formats</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/docs/engineering-ecosystem/common-genomics-tools">Common Genomics Tools</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/docs/engineering-ecosystem/end-to-end-example">Worked Example</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist" aria-expanded="false" href="/docs/advanced-topics/">Advanced Topics</a><button aria-label="Toggle the collapsible sidebar category &#x27;Advanced Topics&#x27;" type="button" class="clean-btn menu__caret"></button></div></li></ul></nav></div></aside><main class="docMainContainer_TCnq"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_DM6M"><div class="docItemContainer_vinB"><article><nav class="theme-doc-breadcrumbs breadcrumbsContainer_Xlws" aria-label="breadcrumbs"><ul class="breadcrumbs" itemscope="" itemtype="https://schema.org/BreadcrumbList"><li class="breadcrumbs__item"><a class="breadcrumbs__link" href="/">🏠</a></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item"><a class="breadcrumbs__link" itemprop="item" href="/docs/engineering-ecosystem/"><span itemprop="name">Engineering Ecosystem</span></a><meta itemprop="position" content="1"></li><li itemscope="" itemprop="itemListElement" itemtype="https://schema.org/ListItem" class="breadcrumbs__item breadcrumbs__item--active"><span class="breadcrumbs__link" itemprop="item name">Genomic File Formats</span><meta itemprop="position" content="2"></li></ul></nav><div class="tocCollapsible_jdIR theme-doc-toc-mobile tocMobile_TmEX"><button type="button" class="clean-btn tocCollapsibleButton_Fzxq">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>Genomic File Formats</h1></header><p>When working with genomics data for the first time, many engineers are surprised to find that genomics file types are quite simple in structure. In fact, a majority of universal file formats such as <a href="https://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener noreferrer">SAM/BAM</a>, <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" target="_blank" rel="noopener noreferrer">VCF</a>, <a href="https://uswest.ensembl.org/info/website/upload/bed.html" target="_blank" rel="noopener noreferrer">BED</a>, <a href="https://uswest.ensembl.org/info/website/upload/gff.html" target="_blank" rel="noopener noreferrer">GFF/GTF</a> and many other custom formats are simply tab-delimited text files with some form of compression. Mastery of Unix command line tools such as grep, less, cut, join, sort, sed, and awk are essential to working effectively in this domain.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="compression-and-bgzf">Compression and BGZF<a class="hash-link" href="#compression-and-bgzf" title="Direct link to heading">​</a></h2><p>The BGZF compression technique and associated <a href="http://www.htslib.org/doc/bgzip.html" target="_blank" rel="noopener noreferrer">bgzip</a>/<a href="https://www.htslib.org/doc/tabix.html" target="_blank" rel="noopener noreferrer">tabix</a> tools were developed as general purpose utilities to enable random access lookup in compressed tab-delimited text files. By default, the gzip compression algorithm creates a single stream of data---accessing information in the middle of that compressed stream requires decompressing the stream from the beginning. Genomic files tend to be substantial in size, so decompressing an entire file to retrieve information is a waste of time and resources. The BGZF compression technique solves this problem by creating multiple gzip streams as contiguous &quot;blocks&quot; and storing the offsets to each block in an associated <strong>index file</strong>. In this case, you can retrieve a subset of information from the file by seeking past all of the irrelevant BGZF blocks (avoiding the associated computational overhead) and only decompressing the block that contains the information you want.</p><p>Consider a single, bgzipped BAM file with the name Sample.bam. This file will typically be accompanied by an index called Sample.bam.bai, which contains offsets to the BGZF blocks. These two files are often treated as a pair.</p><p>For more information, see &quot;The BGZF compression format&quot; section in <a href="http://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener noreferrer">the SAM file specification</a>.</p><h2 class="anchor anchorWithStickyNavbar_mojV" id="comparison-with-gzip">Comparison with Gzip<a class="hash-link" href="#comparison-with-gzip" title="Direct link to heading">​</a></h2><p>Below is a simple comparison of gzip versus bgzip to illustrate the trade-off. Note that while gzip achieves a slight edge in compression ratio, the bgzip/tabix pair is vastly superior in lookup time. This non-linear trade-off between compression and random-access speed is why BGZF is pervasive in the field of computational genomics.</p><p><img loading="lazy" src="/assets/images/7.1-GZIP-BGZF-5c23f2c8afaa746f11cfc9e219652321.jpg" width="1500" height="620" class="img_E7b_"></p><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token assign-left variable" style="color:#36acaa">GENCODE_GTF</span><span class="token operator" style="color:#393A34">=</span><span class="token string" style="color:#e3116c">&quot;ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_32/gencode.v32.chr_patch_hapl_scaff.annotation.gtf.gz&quot;</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">curl</span><span class="token plain"> </span><span class="token string" style="color:#e3116c">&quot;</span><span class="token string variable" style="color:#36acaa">$GENCODE_GTF</span><span class="token string" style="color:#e3116c">&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">                        </span><span class="token comment" style="color:#999988;font-style:italic"># Download GTF</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> gunzip </span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">                                 </span><span class="token comment" style="color:#999988;font-style:italic"># Decompress</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> </span><span class="token function" style="color:#d73a49">grep</span><span class="token plain"> -v </span><span class="token string" style="color:#e3116c">&quot;^#&quot;</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">\</span><span class="token plain">                           </span><span class="token comment" style="color:#999988;font-style:italic"># Remove header lines</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">  </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> </span><span class="token function" style="color:#d73a49">sort</span><span class="token plain"> -k1,1 -k4,4n </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> gencode.v32.all.gtf  </span><span class="token comment" style="color:#999988;font-style:italic"># Sort by chromosome name then genomic start lcocation (numerically)</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">gzip</span><span class="token plain"> -c gencode.v32.all.gtf </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> gencode.v32.all.gtf.gzipped</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">bgzip -c gencode.v32.all.gtf </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> gencode.v32.all.gtf.bgzipped</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">ls</span><span class="token plain"> -lah</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Permissions Size User    Date Modified Name</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># .rw-r--r--  1.4G cmcleod  8 May 23:31  gencode.v32.all.gtf</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># .rw-r--r--   61M cmcleod  8 May 23:32  gencode.v32.all.gtf.bgzipped</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># .rw-r--r--   48M cmcleod  8 May 23:32  gencode.v32.all.gtf.gzipped</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token plain">tabix -p gff gencode.v32.all.gtf.gzipped</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># [tabix] was bgzip used to compress this file? gencode.v32.all.gtf.gzipped</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">tabix -p gff gencode.v32.all.gtf.bgzipped</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">time</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain">zgrep </span><span class="token string" style="color:#e3116c">&quot;gene_name=NOTCH1&quot;</span><span class="token plain"> gencode.v32.all.gtf.gzipped </span><span class="token operator" style="color:#393A34">&amp;&gt;</span><span class="token plain"> /dev/null</span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># ( zgrep &quot;gene_name=NOTCH1&quot; gencode.v32.all.gtf.gzipped &amp;&gt; /dev/null; )  20.60s user 0.05s system 99% cpu 20.693 total</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">time</span><span class="token plain"> </span><span class="token punctuation" style="color:#393A34">(</span><span class="token plain">tabix gencode.v32.all.gtf.bgzipped chr9:136,496,070-136,545,786 </span><span class="token operator" style="color:#393A34">&amp;&gt;</span><span class="token plain"> /dev/null</span><span class="token punctuation" style="color:#393A34">)</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># ( tabix gencode.v32.all.gtf.bgzipped chr9:136,496,070-136,545,786 &amp;&gt; /dev/nul)  0.01s user 0.00s system 83% cpu 0.018 total</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="fastq-files">FASTQ Files<a class="hash-link" href="#fastq-files" title="Direct link to heading">​</a></h2><p>For most workflows in computational genomics, the input files are in the FASTQ format. <a href="https://en.wikipedia.org/wiki/FASTQ_format" target="_blank" rel="noopener noreferrer">The dedicated Wikipedia page</a> for the FASTQ file is quite good, so we recommend you take a look before starting this section. As a reminder, this guide focuses on Illumina-produced FASTQ files, though the concepts are generally applicable to most sequencers today.</p><p>Technically speaking, FASTQ files are not a direct product of sequencers as many have their own intermediate representation of the reads (e.g. <a href="https://www.illumina.com/informatics/sequencing-data-analysis/sequence-file-formats.html" target="_blank" rel="noopener noreferrer">BCL files</a> for Illumina sequencers) that need to be translated into FASTQ files (using a tool like <a href="https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html" target="_blank" rel="noopener noreferrer">bcl2fastq</a>). In practice, many sequencing cores do this translation step before sending the data to the computational engineers downstream.</p><p>A FASTQ file may come as either a single file or an inseparable pair of files for single-end or paired-end sequencing respectively. A FASTQ file contains many reads, which, as noted before, are fragments of DNA/RNA read by the sequencer. Each read in the file has the following four-line structure separated by newlines.</p><p><img loading="lazy" src="/assets/images/6.1-FastQ-e36a740251aef159f725a01cc9d6757c.jpg" width="1500" height="772" class="img_E7b_"></p><table><thead><tr><th><strong>Line Number</strong></th><th><strong>Line Description</strong></th></tr></thead><tbody><tr><td>1</td><td>The sequence name, which on modern Illumina sequencers, this is a coordinate system dictating where the read originated from on the flow cell</td></tr><tr><td>2</td><td>The sequence of the DNA fragment</td></tr><tr><td>3</td><td>A single + delimiter character, often referred to as the &quot;plus line&quot;</td></tr><tr><td>4</td><td>The associated quality scores, one per nucleotide coordinating with line 2; generally an ASCII encoded probability score of incorrectness</td></tr></tbody></table><p>Random access within FASTQ files is not typical, so they are gzipped (not bgzipped) to save space. Conventionally, FASTQ file names indicate which read the files contains (e.g. Sample.fastq.gz for single-end sequencing or Sample_R1.fastq.gz and Sample_R2.fastq.gz in paired-end sequencing where _R1 stands for &quot;read one(s)&quot; and _R2 stands for &quot;read two(s)&quot;).</p><p>Note that in the case of paired-end sequencing, it is crucial that each of the FASTQ files list their reads in same order. If even one read is deleted from either file, the entire read pairing will be off, which will have disastrous results during the alignment phase. To catch common formatting errors in single-end or paired-end FASTQ files, consider using <a href="https://github.com/stjude/fqlib" target="_blank" rel="noopener noreferrer">fqlib</a> (specifically, the lint subcommand).</p><p>Below is an example of the first four lines of a pair of FASTQ files generated by fq generate and validated by fq lint.</p><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token plain">fq generate --n-records </span><span class="token number" style="color:#36acaa">2</span><span class="token plain"> Sample_R1.fastq.gz Sample_R2.fastq.gz</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token function" style="color:#d73a49">gzip</span><span class="token plain"> -dc Sample_R1.fastq.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># @fqlib5:334:VVDJXFE:4:5:7053:1012/1</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># CCTTTTCCCGCAGTCGTCAGCAGTAAGTGTGCGACCGGTAGTTCAAAAGGGGAATATCACCCGCTATTTTGCGAATACTAGAGCCTCGGTTCACGCAAGCA</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># +</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># GFGDDB@CJCHDBDBDB@JACCIE@J@GD@FHFADAIEA@@ABBEADHBFH@CIEFBBHGCHEJHEAIHAAIHBGEH@EDGAEEC@FGEABAHE@FAI@IC</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># @fqlib5:334:VVDJXFE:7:33:2310:7985/1</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># ACGTGGCCGTCCTTTTGCCAGATATCGGTAAGAGAGTTCTAGCTAAGATAATATCAATCCGCGAATGTCAGAGGGAGTGTTTCCCTTCCGGGGAAGCAAAT</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># +</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># GDGE@DBJJCBHCGFEBHDGEI@JA@GHEHA@CCJGCGGDJGBACIBGECDDBGBHI@GBAEBBJD@@BAFBHAHEHJCBGFGCA@GJ@IAAIJDHG@IDC</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token function" style="color:#d73a49">gzip</span><span class="token plain"> -dc Sample_R2.fastq.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># @fqlib5:334:VVDJXFE:4:5:7053:1012/2</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># ATGCCAATGGGAAGCTCGGGTAGTTCTTTCATCATGGTAATAAATCGACCGAACAACCGTTCAGGGCCAGATGAAAAGGGCCTGGCGTCTGGCACAGACCC</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># +</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># FGDFEHBCJFDFCFHEHFGEEADCBEF@JJD@HEGE@@AIIIDHGHADEAHA@D@DCAGBFH@HBDGICI@ICAAGJBCDDGEFJ@JBHGJFACEECJEJJ</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># @fqlib5:334:VVDJXFE:7:33:2310:7985/2</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># GCCCGTAGCAATACTCCTTGCGGGGACGATTATGGCGTTAATCTGATATCTCAAGCAGTAGTGGGGCTATACACTTGCGCCGTAGTCGAGCGGTTTGTAGT</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># +</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># DIGCJJHDCEDDDAGH@FFGHFEAB@JDA@@HH@AJJIHJCA@JFHGAHA@CCGJDF@BJ@AH@B@BH@JCCEIDCIFCED@E@D@II@HBJIADG@EFDB</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><div class="codeBlockContainer_I0IT language-bash theme-code-block"><div class="codeBlockContent_wNvx" style="color:#393A34;background-color:#f6f8fa"><pre tabindex="0" class="prism-code language-bash codeBlock_jd64 thin-scrollbar"><code class="codeBlockLines_mRuA"><span class="token-line" style="color:#393A34"><span class="token plain">fq lint Sample_R1.fastq.gz Sample_R2.fastq.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Exit code is 0, which means all went well.</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Simulate a missing quality score, which would be a malformed file.</span><span class="token plain"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">zcat Sample_R2.fastq.gz </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> </span><span class="token function" style="color:#d73a49">head</span><span class="token plain"> -n </span><span class="token number" style="color:#36acaa">7</span><span class="token plain"> </span><span class="token operator" style="color:#393A34">|</span><span class="token plain"> </span><span class="token function" style="color:#d73a49">gzip</span><span class="token plain"> -c </span><span class="token operator" style="color:#393A34">&gt;</span><span class="token plain"> Sample_R2.bad.fastq.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain">fq lint Sample_R1.fastq.gz Sample_R2.bad.fastq.gz</span><br></span><span class="token-line" style="color:#393A34"><span class="token plain" style="display:inline-block"></span><br></span><span class="token-line" style="color:#393A34"><span class="token plain"></span><span class="token comment" style="color:#999988;font-style:italic"># Sample_R2.bad.fastq.gz:8:1: [S004] CompleteValidator: Incomplete record: quality is empty</span><br></span></code></pre><button type="button" aria-label="Copy code to clipboard" title="Copy" class="copyButton_eDfN clean-btn"><span class="copyButtonIcons_W9eQ" aria-hidden="true"><svg class="copyButtonIcon_XEyF" viewBox="0 0 24 24"><path d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"></path></svg><svg class="copyButtonSuccessIcon_i9w9" viewBox="0 0 24 24"><path d="M21,7L9,19L3.5,13.5L4.91,12.09L9,16.17L19.59,5.59L21,7Z"></path></svg></span></button></div></div><h2 class="anchor anchorWithStickyNavbar_mojV" id="bam-files">BAM Files<a class="hash-link" href="#bam-files" title="Direct link to heading">​</a></h2><p>The <strong>Sequence Alignment/Map</strong> (or SAM) format is the most common output file format after the read alignment. Typically, these files will be encoded in a binary format, bgzipped, indexed, at which point the file is referred to as a Binary Alignment/Map Format or BAM file. Each BAM file contains a header (typically used for storing metadata) followed by millions of reads, stored as one read per line. Each read contains all of the information available in the FASTQ file (read name, sequence, associated quality score), as well as additional information added by the aligner, including where the read maps to the reference genome, whether the read was produced by PCR or optical duplication, and the quality of the alignment assigned by the mapping software.</p><p>Ultimately, a SAM file is a tab-delimited text file that could be piped into standard Unix tools for processing. BAM files can be treated in much the same way if passed through samtools view first. Learn more about it in the associated <a href="https://samtools.github.io/hts-specs/SAMv1.pdf" target="_blank" rel="noopener noreferrer">documentation</a>.</p><p><img loading="lazy" src="/assets/images/6.2-BAM-2dd71d8e7b3ee35633667bf0ff8e9cd4.jpg" width="1000" height="959" class="img_E7b_"></p><h2 class="anchor anchorWithStickyNavbar_mojV" id="vcf-files">VCF Files<a class="hash-link" href="#vcf-files" title="Direct link to heading">​</a></h2><p>The <strong>Variant Call Format</strong> (or VCF) is the most common format for storing variants detected in next-generation sequencing data. Typically, these files will be bgzipped and indexed. Each VCF file contains a header (typically used for storing metadata) and a list of variants. Minimally, each variant contains information about the where the variant lives in the genome and a definition of the change caused by the variant. It also supports storing information about the cohort samples and which samples contain which variants (effectively a <!-- -->[num_variants]<!-- --> X <!-- -->[num_samples]<!-- --> matrix with information on every sample, variant pair.)</p><p>Concretely, the minimal VCF looks like this.</p><table><thead><tr><th><strong>CHROM</strong></th><th><strong>POS</strong></th><th><strong>ID</strong></th><th><strong>REF</strong></th><th><strong>ALT</strong></th><th><strong>QUAL</strong></th><th><strong>FILTER</strong></th><th><strong>INFO</strong></th></tr></thead><tbody><tr><td>chr1</td><td>114713908</td><td>rs11554290</td><td>T</td><td>A</td><td>.</td><td>.</td><td></td></tr><tr><td>chr17</td><td>7683718</td><td>.</td><td>C</td><td>CT</td><td>.</td><td>.</td><td></td></tr></tbody></table><p>These fields describe:</p><ul><li><p><strong>Variant Location.</strong> Typically, the chromosome, position, reference base(s), and alternative base(s) fields are used to create an &quot;address&quot; of the variant in the genome using the form chrom-pos-ref-alt.</p></li><li><p><strong>Variant Identifier.</strong> The ID field is used to store one or more unique identifiers for the variant. Commonly, the rs numbers from <a href="https://www.ncbi.nlm.nih.gov/snp/" target="_blank" rel="noopener noreferrer">dbSNP</a> are used.</p></li><li><p><strong>Quality of the Variant.</strong> A measure of confidence in the called variant or lack thereof. This field follows the common practice of encoding qualities as a Phred-scaled quality score with the formula . A score of 10 means there is a 1 in 10 chance of the call being incorrect, 20 means a 1 in 100 chance of the call being incorrect, and so on. If unknown, the value should be the missing value, a period.</p></li><li><p><strong>Filter Status.</strong> This denotes the status of whether the variant passed all quality filters from the caller. If the variant passes, the value will have a value of PASS. If the variant fails any filters, the value will be a semi-colon separated list of codes for the filters. If unknown, the value should be the missing value.</p></li><li><p><strong>Additional Information.</strong> A field which can contain an arbitrary number of key-value pairs separated by semicolons of the form /<key></key>=/<value1></value1>[,/<value2></value2>]<!-- -->. While there are some reserved keys specified in the documentation, anything can be denoted here. If there is no additional information, this field is empty.</p></li></ul><p>As alluded to earlier, the VCF format is flexible in that it <em>may</em> specify one or more samples and indicate whether the sample contains any of the variants specified in the file. A simplified version of this would be something like:</p><p><strong>Chromosome</strong> <strong>Position</strong> <strong>ID</strong> <strong>REF</strong> <strong>ALT</strong> <strong>Sample1</strong> <strong>Sample2</strong> <strong>Sample3</strong></p><hr><p>chr1 114713908 rs11554290 T A YES NO NO
chr17 7683718 . C CT NO YES YES</p><p>However, in practice this is complicated by the fact that variants can have various genotypes (homozygous reference allele, homozygous alternate allele, or heterozygous) and evidence depth. For more information, see the &quot;Genotype fields&quot; section of the <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" target="_blank" rel="noopener noreferrer">VCF specification</a>.</p><p>Ultimately, a VCF file is just a tab-delimited text file that could be piped into standard Unix tools for processing. Learn more about it <a href="https://samtools.github.io/hts-specs/VCFv4.3.pdf" target="_blank" rel="noopener noreferrer">here</a>.</p></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/facebook/docusaurus/tree/main/packages/create-docusaurus/templates/shared/docs/04-engineering-ecosystem/01-genomic-file-formats.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_dcUD" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_foO9"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/docs/engineering-ecosystem/"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Engineering Ecosystem</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/docs/engineering-ecosystem/common-genomics-tools"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Common Genomics Tools</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_cNA8 thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#compression-and-bgzf" class="table-of-contents__link toc-highlight">Compression and BGZF</a></li><li><a href="#comparison-with-gzip" class="table-of-contents__link toc-highlight">Comparison with Gzip</a></li><li><a href="#fastq-files" class="table-of-contents__link toc-highlight">FASTQ Files</a></li><li><a href="#bam-files" class="table-of-contents__link toc-highlight">BAM Files</a></li><li><a href="#vcf-files" class="table-of-contents__link toc-highlight">VCF Files</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container container-fluid"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Guide</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/docs/biological-foundations">Biological Foundations</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/genomic-variation">Genomic Variation</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/next-generation-sequencing">Next-generation Sequencing</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/engineering-ecosystem">Engineering Ecosystem</a></li><li class="footer__item"><a class="footer__link-item" href="/docs/advanced-topics">Advanced Topics</a></li></ul></div><div class="col footer__col"><div class="footer__title">Socials</div><ul class="footer__items"><li class="footer__item"><a href="https://github.com/learngenomics/learngenomics.dev" target="_blank" rel="noopener noreferrer" class="footer__link-item">GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_I5OW"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2022 learngenomics.dev and associated contributors. Built with Docusaurus.</div></div></div></footer></div>
<script src="/assets/js/runtime~main.fe122cfd.js"></script>
<script src="/assets/js/main.585ef083.js"></script>
</body>
</html>